---
title: "FIxGeMoMa for genome v2"
author: "Ariana S Huffmyer"
date: "2/9/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
#Load libraries
library(tidyverse)
library(ape)

#Load Mcap gene gff
#Mcap.gff <- read.gff("Mcap2020/Data/TagSeq/Montipora_capitata_HIv2.genes.gff3", na.strings = c("."), GFF3 = TRUE)

New.gff <- read.csv(file="Mcap2020/Data/TagSeq/Montipora_capitata_HIv2.genes.gff3", header=FALSE, sep="\t") 

Mcap.gff <- read.csv(file="Mcap2020/Data/TagSeq/Mcap.GFFannotation.gff", header=FALSE, sep="\t")



###### These formats are super different - we need to figure out what to do with this GFF file to match the format from before? 
#The new format has liftoff for gene prediction... 


#rename columns
colnames(New.gff) <- c("scaffold", "Gene.Predict", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")

#remove excess information from GeMoMa so it matches AUGUSTUS
New.gff$gene <- gsub("_cds[0123456789]*;", ".t1;", New.gff$gene) #change the second part of the GeMoMa genes from cds11 to t.1 to match Augustus
Mcap.gff$gene <-sub("(;[^;]+);.*", "\\1", Mcap.gff$gene) #remove everything after the second ; in the gene column
Mcap.gff$gene <- gsub("Parent=", "  gene_id ", Mcap.gff$gene) #remove ID= from GeMoMa genes

#If id ==CDS replace ID= with transcript_id, else replace with nothing
Mcap.gff <- Mcap.gff %>% 
  mutate(gene = ifelse(Gene.Predict %in% c("GeMoMa") & 
                         id == "CDS" ,  
                       gsub("ID=", "transcript_id ", gene), gsub("", "", gene)))

#If id ==intron replace ID= with transcript_id, else replace with nothing
Mcap.gff <- Mcap.gff %>% 
  mutate(gene = ifelse(Gene.Predict %in% c("GeMoMa") & 
                         id == "intron" ,  
                       gsub("ID=", "transcript_id ", gene), gsub("ID=", "", gene)))

#If id ==gene remove everything after the ; else replace with nothing
Mcap.gff <- Mcap.gff %>% 
  mutate(gene = ifelse(Gene.Predict %in% c("GeMoMa") & 
                         id == "gene" ,  
                       gsub(";.*", "", gene), gsub("ID=", "", gene)))

# sub to add quotes around the transcript name
Mcap.gff$gene <- gsub("transcript_id ", "transcript_id \"", Mcap.gff$gene) 
Mcap.gff$gene <- gsub(";", "\";", Mcap.gff$gene) 

#add quotes before the gene_id
Mcap.gff$gene <- gsub("gene_id ", "gene_id \"", Mcap.gff$gene) 

#If id ==CDS add "; at the end else replace with nothing
Mcap.gff <- Mcap.gff %>% 
  mutate(gene = ifelse(id == "CDS" ,  
                         paste0(gene, "\";"), paste0(gene, "")))

#If id ==CDS add "; at the end else replace with nothing
Mcap.gff <- Mcap.gff %>% 
  mutate(gene = ifelse(id == "intron" ,  
                       paste0(gene, "\";"), paste0(gene, "")))

#save file
write.table(Mcap.gff, file="Mcap2020/Data/TagSeq/Mcap.GFFannotation.fixed.gff", sep="\t", 
            col.names = FALSE, row.names=FALSE, quote=FALSE)
```