---
title: "WGCNA gene expression"
author: "Ariana S Huffmyer"
date: "8/13/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

WGCNA analysis of TagSeq dataset for Montipora capitata developmental time series (2020). These scripts are based on scripts from Erin Chille's master's thesis at the University of Rhode Island (https://github.com/echille).  

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# The following setting is important, do not omit.
options(stringsAsFactors = FALSE) #Set Strings to character
```

Load required libraries.  

```{r}
if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse') 
if ("genefilter" %in% rownames(installed.packages()) == 'FALSE') install.packages('genefilter') 
if ("DESeq2" %in% rownames(installed.packages()) == 'FALSE') install.packages('DESeq2') 
if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer') 
if ("WGCNA" %in% rownames(installed.packages()) == 'FALSE') install.packages('WGCNA') 
if ("flashClust" %in% rownames(installed.packages()) == 'FALSE') install.packages('flashClust') 
if ("gridExtra" %in% rownames(installed.packages()) == 'FALSE') install.packages('gridExtra') 
if ("ComplexHeatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('ComplexHeatmap') 
if ("goseq" %in% rownames(installed.packages()) == 'FALSE') install.packages('goseq') 
if ("dplyr" %in% rownames(installed.packages()) == 'FALSE') install.packages('dplyr') 
if ("clusterProfiler" %in% rownames(installed.packages()) == 'FALSE') install.packages('clusterProfiler') 
if ("pheatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('pheatmap') 
if ("magrittr" %in% rownames(installed.packages()) == 'FALSE') install.packages('magrittr') 
if ("vegan" %in% rownames(installed.packages()) == 'FALSE') install.packages('vegan') 
if ("factoextra" %in% rownames(installed.packages()) == 'FALSE') install.packages('factoextra') 

library("tidyverse")
library("genefilter")
library("DESeq2")
library("RColorBrewer")
library("WGCNA")
library("flashClust")
library("gridExtra")
library("ComplexHeatmap")
library("goseq")
library("dplyr")
library("clusterProfiler")
library("pheatmap")
library("magrittr")
library("vegan")
library("factoextra")

```

# **Data input and filtering**  
 
Import the data files.  

```{r}
#load metadata sheet with sample name and developmental stage information
metadata <- read.csv("Mcap2020/Data/TagSeq/Sample_Info.csv", header = TRUE, sep = ",")
head(metadata)

#load gene count matrix generated from cluster computation
gcount <- as.data.frame(read.csv("Mcap2020/Output/TagSeq/Mcapitata_gene_count_matrix.csv", row.names="gene_id"), colClasses = double)
head(gcount)

#remove metamorphosed recruit 1 because this was the timepoint that we only had one sample for: D36, AH23
#gcount<-gcount[ , !(names(gcount) %in% c("AH23"))]
gcount<-gcount[ , !(names(gcount) %in% c("AH23_S54_L002.gtf"))]

metadata <- metadata%>%
  filter(!sample_id=="AH23")
```

Check that there are no genes with 0 counts across all samples.

```{r}
nrow(gcount)
gcount<-gcount %>%
     mutate(Total = rowSums(.[, 1:38]))%>%
    filter(!Total==0)%>%
    select(!Total)
nrow(gcount)
```
We had 63,000 genes, which was filtered down to 39,495 by removing genes with row sums of 0.  

Conduct data filtering, this includes:  

*pOverA*: Specifying the minimum count for a proportion of samples for each gene. Here, we are using a pOverA of 0.07. This is because we have 38 samples with a minimum of n=3 samples per lifestage. Therefore, we will accept genes that are present in 3/38 = 0.07 of the samples because we expect different expression by life stage. We are further setting the minimum count of genes to 10, such that 7% of the samples must have a gene count of >10 in order for the gene to remain in the data set.  

Filter in the package "genefilter". Pre-filtering our dataset to reduce the memory size dataframe, increase the speed of the transformation and testing functions, and improve quality of statistical analysis by removing low-coverage counts. Removed counts could represent outliers in the data and removing these improves sensitivity of statistical tests.   

```{r}
filt <- filterfun(pOverA(0.07,5))

#create filter for the counts data
gfilt <- genefilter(gcount, filt)

#identify genes to keep by count filter
gkeep <- gcount[gfilt,]

#identify genes to keep by count filter
gkeep <- gcount[gfilt,]

#identify gene lists
gn.keep <- rownames(gkeep)

#gene count data filtered in PoverA, P percent of the samples have counts over A
gcount_filt <- as.data.frame(gcount[which(rownames(gcount) %in% gn.keep),])

#How many rows do we have before and after filtering?
nrow(gcount) #Before
nrow(gcount_filt) #After
```

Before filtering, we had 39,495 genes. After filtering for pOverA, we have approximately 15,797 genes. This indicates that there were many genes present in <7% of samples at <5 counts per gene.  

In order for the DESeq2 algorithms to work, the SampleIDs on the metadta file and count matrices have to match exactly and in the same order. The following R clump will check to make sure that these match. Should return TRUE.  
```{r}
#Checking that all row and column names match. Should return "TRUE"
all(rownames(metadata$sample_id) %in% colnames(gcount_filt))
all(rownames(metadata$sample_id) == colnames(gcount_filt)) 
```

Match metadata sample id's.  
```{r}
result<-sub("_.*", "", colnames(gcount_filt)) #keep characters before _
#result2<-sub(".*AH", "", result) #keep characters after X
colnames(gcount_filt)<-result #keep just sample number 
```

Display current order of metadata and gene count matrix.  
```{r}
metadata$sample_id
colnames(gcount_filt)
```

Order metadata the same as the column order in the gene matrix.  
```{r}
list<-colnames(gcount_filt)
list<-as.factor(list)

metadata$sample_id<-as.factor(metadata$sample_id)

# Re-order the levels
metadata$sample_id <- factor(as.character(metadata$sample_id), levels=list)
# Re-order the data.frame
metadata_ordered <- metadata[order(metadata$sample_id),]
metadata_ordered$sample_id

metadata_ordered<-metadata_ordered%>%
  select(sample_id, lifestage, hpf, code)
```

Reorder the lifestage factor.  
```{r}
head(metadata_ordered)
levels(as.factor(metadata_ordered$lifestage))

metadata_ordered$lifestage<-as.factor(metadata_ordered$lifestage)

metadata_ordered$lifestage<-fct_relevel(metadata_ordered$lifestage, "Egg (1 hpf)", "Embryo (5 hpf)", "Embryo (38 hpf)", "Embryo (65 hpf)", "Larvae (93 hpf)", "Larvae (163 hpf)", "Larvae (231 hpf)", "Metamorphosed Polyp (231 hpf)", "Attached Recruit (183 hpf)", "Attached Recruit (231 hpf)")

levels(as.factor(metadata_ordered$lifestage))
```

Plot the sequence counts after trimming in each sample to confirm that sequence depth is not different between lifestages.  
```{r}
reads<-read.csv("Mcap2020/Output/TagSeq/clean_seqs_stats.csv")

#match in lifestage information 
reads$lifestage<-metadata$lifestage[match(reads$Sample, metadata$sample_id)]
head(reads)

reads$lifestage<-as.factor(reads$lifestage)
levels(reads$lifestage)

reads$lifestage<-fct_relevel(reads$lifestage, "Egg (1 hpf)", "Embryo (5 hpf)", "Embryo (38 hpf)", "Embryo (65 hpf)", "Larvae (93 hpf)", "Larvae (163 hpf)", "Larvae (231 hpf)", "Metamorphosed Polyp (231 hpf)", "Attached Recruit (183 hpf)", "Attached Recruit (231 hpf)")

reads %>% ggplot(., aes(x = lifestage, y = M.Seqs)) +
  geom_boxplot()
```
No pattern between reads and life stage. Note that "NA" here refers to the one metamorpohsed recruit time point that is removed from analysis.  

# **Construct DESeq2 data set**  

Create a DESeqDataSet design from gene count matrix and labels. Here we set the design to look at lifestage to test for any differences in gene expression across timepoints.
```{r}
#Set DESeq2 design
gdds <- DESeqDataSetFromMatrix(countData = gcount_filt,
                              colData = metadata_ordered,
                              design = ~lifestage)
```

First we are going to log-transform the data using a variance stabilizing transforamtion (VST). This is only for visualization purposes. Essentially, this is roughly similar to putting the data on the log2 scale. It will deal with the sampling variability of low counts by calculating within-group variability (if blind=FALSE). Importantly, it does not use the design to remove variation in the data, and so can be used to examine if there may be any variability do to technical factors such as extraction batch effects.

To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4.

Chunk should return TRUE if <4.  
```{r}
SF.gdds <- estimateSizeFactors(gdds) #estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than 4 for us to use vst
print(sizeFactors(SF.gdds)) #View size factors

all(sizeFactors(SF.gdds)) < 4
```

All size factors are less than 4, so we can use VST transformation.  

```{r}
gvst <- vst(gdds, blind=FALSE) #apply a variance stabilizing transformation to minimize effects of small counts and normalize wrt library size
head(assay(gvst), 3) #view transformed gene count data for the first three genes in the dataset.  
```

# **Conduct PERMANOVA by developmental time point**  

Export data for PERMANOVA test.  
```{r}
test<-t(assay(gvst)) #export as matrix
test<-as.data.frame(test)

#add category columns
test$Sample<-rownames(test)
test$Lifestage<-metadata$lifestage[match(test$Sample, metadata$sample_id)]
```

Build PERMANOVA model.  
```{r}
scaled_test <-prcomp(test[c(1:15797)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test)

# scale data
vegan <- scale(test[c(1:15797)])

# PerMANOVA 
permanova<-adonis2(vegan ~ Lifestage, data = test, method='eu')
permanova
```

Gene expression is significantly different between temperatures and date.  

# **Examine PCA and sample distances**  

Plot a heatmap to sample to sample distances  

```{r}
gsampleDists <- dist(t(assay(gvst))) #calculate distance matix
gsampleDistMatrix <- as.matrix(gsampleDists) #distance matrix
rownames(gsampleDistMatrix) <- colnames(gvst) #assign row names
colnames(gsampleDistMatrix) <- NULL #assign col names
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255) #assign colors

save_pheatmap_pdf <- function(x, filename, width=7, height=7) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}

pht<-pheatmap(gsampleDistMatrix, #plot matrix
         clustering_distance_rows=gsampleDists, #cluster rows
         clustering_distance_cols=gsampleDists, #cluster columns
         col=colors) #set colors

save_pheatmap_pdf(pht, "Mcap2020/Figures/TagSeq/pheatmap.pdf")
```

Plot a PCA of samples by lifestages  

```{r}
gPCAdata <- plotPCA(gvst, intgroup = c("lifestage"), returnData=TRUE, ntop=15797) #use ntop to specify all genes

percentVar <- round(100*attr(gPCAdata, "percentVar")) #plot PCA of samples with all data

allgenesfilt_PCA <- ggplot(gPCAdata, aes(PC1, PC2, shape=lifestage)) + 
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  scale_shape_manual(values = c("Egg (1 hpf)"=17, "Attached Recruit (231 hpf)"=1, "Embryo (5 hpf)"=10, "Embryo (38 hpf)"=5, "Embryo (65 hpf)"=14, "Larvae (93 hpf)"=11, "Larvae (163 hpf)"=6, "Larvae (231 hpf)"=2, "Metamorphosed Polyp (231 hpf)"=8, "Attached Recruit (183 hpf)"=3)) +
  #xlim(-40,40)+ 
  #ylim(-40,40)+
  coord_fixed()+
  theme_bw() + #Set background color
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) # + #Set the plot background
  #theme(legend.position = ("none")) #set title attributes
allgenesfilt_PCA
ggsave("Mcap2020/Figures/TagSeq/allgenesfilt-PCA.pdf", allgenesfilt_PCA, width=11, height=8)
```

Generate a PCA for visualization/presentations.  

```{r}
levels(gPCAdata$lifestage)

allgenesfilt_PCA_visual <- 
  ggplot(data=gPCAdata, aes(PC1, PC2, shape=lifestage, colour=lifestage)) + 
  geom_point(size=5) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  
  scale_shape_manual(name="Lifestage", values = c("Egg (1 hpf)"=17, "Embryo (5 hpf)"=16, "Embryo (38 hpf)"=15, "Embryo (65 hpf)"=17, "Larvae (93 hpf)"=15, "Larvae (163 hpf)"=16, "Larvae (231 hpf)"=17, "Metamorphosed Polyp (231 hpf)"=16, "Attached Recruit (183 hpf)"=15, "Attached Recruit (231 hpf)"=16)) +
  
  scale_colour_manual(name="Lifestage", values = c("Egg (1 hpf)"="#8C510A", "Embryo (5 hpf)"="#DFC27D", "Embryo (38 hpf)"="#DFC27D", "Embryo (65 hpf)"="#DFC27D", "Larvae (93 hpf)"="#80CDC1", "Larvae (163 hpf)"="#80CDC1", "Larvae (231 hpf)"="#80CDC1", "Metamorphosed Polyp (231 hpf)"="#003C30", "Attached Recruit (183 hpf)"="#BA55D3", "Attached Recruit (231 hpf)"="#BA55D3")) +
  #xlim(-40,40)+ 
  #ylim(-40,40)+
  coord_fixed()+
  theme_classic() + #Set background color
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) # + #Set the plot background
  #theme(legend.position = ("none")) #set title attributes
allgenesfilt_PCA_visual

ggsave("Mcap2020/Figures/TagSeq/allgenesfilt-PCA_visual.pdf", allgenesfilt_PCA_visual, width=11, height=8)
```

# **WGCNA analysis using Dynamic Tree Cut**  

Data are analyzed using dynamic tree cut approach, our data set is not large enough to have to use blockwiseModules to break data into "blocks". This code uses step by step network analysis and module detection based on scripts from https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/index.html. 

Transpose the filtered gene count matrix so that the gene IDs are rows and the sample IDs are columns.
```{r}
datExpr <- as.data.frame(t(assay(gvst))) #transpose to output to a new data frame with the column names as row names. And make all data numeric
```

Check for genes and samples with too many missing values with goodSamplesGenes. There shouldn't be any because we performed pre-filtering
```{r}
gsg = goodSamplesGenes(datExpr, verbose = 3)
gsg$allOK #Should return TRUE if not, the R chunk below will take care of flagged data
```

Remove flagged samples if the allOK is FALSE, not used here.  
```{r}
#ncol(datExpr) #number genes before
#if (!gsg$allOK) #If the allOK is FALSE...
#{
# Optionally, print the gene and sample names that are flagged:
#if (sum(!gsg$goodGenes)>0)
#printFlush(paste("Removing genes:", paste(names(datExpr)[!gsg$goodGenes], collapse = ", ")));
#if (sum(!gsg$goodSamples)>0)
#printFlush(paste("Removing samples:", paste(rownames(datExpr)[!gsg$goodSamples], collapse = ", ")));
# Remove the offending genes and samples from the data:
#datExpr = datExpr[gsg$goodSamples, gsg$goodGenes]
#}
#ncol(datExpr) #number genes after
```

Look for outliers by examining tree of samples  
```{r}
sampleTree = hclust(dist(datExpr), method = "average");
# Plot the sample tree: Open a graphic output window of size 12 by 9 inches
# The user should change the dimensions if the window is too large or too small.
pdf("Mcap2020/Figures/TagSeq/outliers.pdf")
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5, cex.axis = 1.5, cex.main = 2)
dev.off()
```
There don't look to be any outliers, so we will move on with business as usual.  

## Network construction and consensus module detection  

### Choosing a soft-thresholding power: Analysis of network topology β  

The soft thresholding power (β) is the number to which the co-expression similarity is raised to calculate adjacency. The function pickSoftThreshold performs a network topology analysis. The user chooses a set of candidate powers, however the default parameters are suitable values.  
```{r, message=FALSE, warning=FALSE}
allowWGCNAThreads()
# # Choose a set of soft-thresholding powers
powers <- c(seq(from = 1, to=19, by=2), c(21:30)) #Create a string of numbers from 1 through 10, and even numbers from 10 through 20
# 
# # Call the network topology analysis function
sft <-pickSoftThreshold(datExpr, powerVector = powers, verbose = 5)
```

Plot the results.  
```{r}
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# # # Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
      xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"));
 text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=cex1,col="red");
# # # this line corresponds to using an R^2 cut-off
 abline(h=0.9,col="red")
# # # Mean connectivity as a function of the soft-thresholding power
 plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
 text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
```
I used a scale-free topology fit index **R^2 of 0.9**. This lowest recommended R^2 by Langfelder and Horvath is 0.8. I chose 0.9 because we want to use the smallest soft thresholding power that maximizes with model fit. It appears that our **soft thresholding power is 5** because it is the lowest power before the R^2=0.9 threshold that maximizes with model fit.  

### Co-expression adjacency and topological overlap matrix similarity  

Save Rdata necessary for analysis
```{r}
save(datExpr, file = "Mcap2020/Output/TagSeq/datExpr.RData")
```

Co-expression similarity and adjacency, using the soft thresholding power 5 and translate the adjacency into topological overlap matrix to calculate the corresponding dissimilarity. I will use a **signed network**: https://peterlangfelder.com/2018/11/25/__trashed/
```{r, }
# #Set up workspace
#getwd() #Display the current working directory
# #If necessary, change the path below to the directory where the data files are stored. "." means current directory. On Windows use a forward slash / instead of the usual \.
#workingDir = ".";
# setwd(WGCNA_dev);
# library(WGCNA) #Load the WGCNA package
options(stringsAsFactors = FALSE) #The following setting is important, do not omit.
enableWGCNAThreads() #Allow multi-threading within WGCNA. 
# 
# #Load the data saved in the first part
adjTOM <- load(file="Mcap2020/Output/TagSeq/datExpr.RData")
adjTOM
# 
# #Run analysis
softPower=5 #Set softPower to 5
adjacency=adjacency(datExpr, power=softPower,type="signed") #Calculate adjacency

TOM= TOMsimilarity(adjacency,TOMType = "signed") #Translate adjacency into topological overlap matrix
#this step can take awhile 

dissTOM= 1-TOM #Calculate dissimilarity in TOM

#save(adjacency, TOM, dissTOM, file = "Mcap2020/Output/TagSeq/adjTOM.RData") #Save 

#save(dissTOM, file = "Mcap2020/Output/TagSeq/dissTOM.RData") #Save 
```

Load in dissTOM file obtained from previous R chunk.    
```{r}
#dissTOM_in <- load(file="Mcap2020/Output/TagSeq/dissTOM.RData") 
#dissTOM_in
```

### Clustering using TOM

Form distance matrix  
```{r}
geneTree=flashClust(as.dist(dissTOM), method="average")
```

We will now plot a dendrogram of genes. Each leaf corresponds to a gene, branches grouping together densely are interconnected, highly co-expressed genes.  
```{r}
pdf(file="Mcap2020/Figures/TagSeq/dissTOMClustering.pdf", width=20, height=20)
plot(geneTree, xlab="", sub="", main= "Gene Clustering on TOM-based dissimilarity", labels= FALSE,hang=0.04)
dev.off()
```

### Module identification using dynamicTreeCut  

Module identification is essentially cutting the branches off the tree in the dendrogram above. We like large modules, so we set the **minimum module size** relatively high, so we will set the minimum size at 20. 

```{r}
minModuleSize = 20
dynamicMods = cutreeDynamic(dendro = geneTree, distM = dissTOM,
deepSplit = TRUE, pamRespectsDendro = FALSE,
minClusterSize = minModuleSize)
table(dynamicMods) #list modules and respective sizes
save(dynamicMods, geneTree, file = "Mcap2020/Output/TagSeq/dyMod_geneTree.RData") #Save to load into RStudio
```
Module 0 is reserved for unassigned genes. The are other modules will be listed largest to smallest. 

   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19 
3954 3662 2934  955  913  765  679  602  591  284   85   74   71   54   51   36   34   29   24 

Load modules calculated from the adjacency matrix.  
```{r}
dyMod_geneTree <- load(file = "Mcap2020/Output/TagSeq/dyMod_geneTree.RData")
dyMod_geneTree
```

Plot the module assignment under the gene dendrogram
```{r}
#dynamicColors = labels2colors(dynamicMods) # Convert numeric labels into colors
dynamicColors=dynamicMods
table(dynamicColors)

pdf(file="Mcap2020/Figures/TagSeq/dissTOMColorClustering.pdf", width=20, height=20)
plotDendroAndColors(geneTree, dynamicColors, "Dynamic Tree Cut", dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05, main = "Gene dendrogram and module colors")
dev.off()
```

### Merge modules with similar expression profiles  

Plot module similarity based on eigengene value 
```{r}
#Calculate eigengenes
MEList = moduleEigengenes(datExpr, colors = dynamicColors, softPower = 5)
MEs = MEList$eigengenes

#Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs)

#Cluster again and plot the results
METree = flashClust(as.dist(MEDiss), method = "average")

pdf(file="Mcap2020/Figures/TagSeq/eigengeneClustering1.pdf", width = 20)
plot(METree, main = "Clustering of module eigengenes", xlab = "", sub = "")
dev.off()
```

**Merge modules with >85% eigengene similarity.** Most studies use somewhere between 80-90% similarity. It looks like most of our modules are highly related so I will use 85% similarity as my merging threshold.
```{r}
MEDissThres= 0.15 #merge modules that are 85% similar

pdf(file="Mcap2020/Figures/TagSeq/eigengeneClustering2.pdf", width = 20)
plot(METree, main = "Clustering of module eigengenes", xlab = "", sub = "")
abline(h=MEDissThres, col="red")
dev.off()

merge= mergeCloseModules(datExpr, dynamicColors, cutHeight= MEDissThres, verbose =3)

mergedColors= merge$colors
mergedMEs= merge$newMEs

pdf(file="Mcap2020/Figures/TagSeq/mergedClusters.pdf", width=20, height=20)
plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c("Dynamic Tree Cut", "Merged dynamic"), dendroLabels= FALSE, hang=0.03, addGuide= TRUE, guideHang=0.05)
dev.off()
```

We had 19 modules before merging.  

Save new colors
```{r}
moduleLabels=mergedColors
moduleColors = mergedColors # Rename to moduleColors
#colorOrder = c("grey", standardColors(50)); # Construct numerical labels corresponding to the colors
#moduleLabels = match(moduleColors, colorOrder)-1;
MEs = mergedMEs;
ncol(MEs) #How many modules do we have now?
```

We have 12 modules after merging with 85% similarity.   

Plot new tree with 12 modules.  
```{r}
#Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs)
#Cluster again and plot the results
pdf(file="Mcap2020/Figures/TagSeq/eigengeneClustering3.pdf")
METree = flashClust(as.dist(MEDiss), method = "average")
MEtreePlot = plot(METree, main = "Clustering of module eigengenes", xlab = "", sub = "")
dev.off()
```

Display table of module gene counts.  

```{r}
table(mergedColors)
```

We have 12 modules with no genes "unassigned". All genes were assigned to a module.  

mergedColors
   1    2    4    6    8   11   12   14   15   16   17   18 
3954 3662  955 1049 2194  156 3599   54   51   36   58   29  

##  Relating modules to developmental stage

### Quantifying module–trait associations

Prepare trait data. Data has to be numeric, so I will substitute time_points and type for numeric values. The "trait" we are considering here is ontogeny.  

Make a dataframe that has a column for each lifestage name and a row for samples. Populate a 1 for samples that match each lifestage and a 0 for samples not matching respective lifestages. 

This process changes lifestages from a categorical variable into a binary variable. This will allow for correlations between mean eigengenes and lifestage.  
 
```{r}
metadata_ordered$num <- c("1")
allTraits <- as.data.frame(pivot_wider(metadata_ordered, names_from = lifestage, values_from = num, id_cols = sample_id))
allTraits[is.na(allTraits)] <- c("0")
rownames(allTraits) <- allTraits$sample_id
datTraits <- allTraits[,c(-1)]
datTraits
```

Define numbers of genes and samples and print.  
```{r}
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)

nGenes
nSamples
```

We have 15,797 genes and 38 samples.  

Generate labels for module eigengenes as numbers.  
```{r}
MEs0 = moduleEigengenes(datExpr, moduleLabels, softPower=5)$eigengenes
MEs = orderMEs(MEs0)
names(MEs)
```

Correlations of traits with eigengenes
```{r}
moduleTraitCor = cor(MEs, datTraits, use = "p");
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples);
Colors=sub("ME","", names(MEs))

moduleTraitTree = hclust(dist(t(moduleTraitCor)), method = "average")

pdf(file="Mcap2020/Figures/TagSeq/ModuleTraitClusterTree.pdf")
plot(moduleTraitTree)
dev.off()
```

Correlations of genes with eigengenes. Calculate correlations between ME's and lifestages. 
```{r}
moduleGeneCor=cor(MEs,datExpr)
moduleGenePvalue = corPvalueStudent(moduleGeneCor, nSamples);
```

Calculate kME values (module membership). 

```{r}
datKME = signedKME(datExpr, MEs, outputColumnName = "kME")
head(datKME)
```

Save module colors and labels for use in subsequent analyses.  

```{r}
save(MEs, moduleLabels, moduleColors, geneTree, file="Mcap2020/Output/TagSeq/NetworkConstruction-stepByStep.RData") 
```

### Plot module-trait associations

Represent module trait correlations as a heatmap 
```{r}
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",signif(moduleTraitPvalue, 1), ")", sep = "")
dim(textMatrix) = dim(moduleTraitCor)
head(textMatrix)

labeledHeatmap(Matrix = moduleTraitCor, xLabels = names(datTraits),  yLabels = names(MEs), ySymbols = names(MEs), cex.lab.y= 0.55, cex.lab.x= 0.55, colors = blueWhiteRed(50), textMatrix = textMatrix, setStdMargins = TRUE, cex.text = 0.25, textAdj = , zlim = c(-1,1), main = paste("Module-trait relationships"))

pdf(file="Mcap2020/Figures/TagSeq/Module-trait-relationships.pdf")
labeledHeatmap(Matrix = moduleTraitCor, xLabels = names(datTraits),  yLabels = names(MEs), ySymbols = names(MEs), cex.lab.y= 0.55, cex.lab.x= 0.55, colors = blueWhiteRed(50), textMatrix = textMatrix, setStdMargins = TRUE, cex.text = 0.25, textAdj = , zlim = c(-1,1), main = paste("Module-trait relationships"))
dev.off()
```

Generate a complex heatmap of module-trait relationships.  

```{r}
#bold sig p-values
#dendrogram with WGCNA MEtree cut-off
#colored y-axis

#Create list of pvalues for eigengene correlation with specific life stages
heatmappval <- signif(moduleTraitPvalue, 1)

#Make list of heatmap row colors
htmap.colors <- names(MEs)
htmap.colors <- gsub("ME", "", htmap.colors)

#lifestage_order<-c("1_Egg", "2_Embryo", "3_Larvae1", "4_Larvae2", "5_Larvae3", "6_Larvae4", "7_Larvae6", "8_MetamorposedRecruit2", "9_CalcifyingRecruit1", "10_CalcifyingRecruit2")

library(dendsort)
row_dend = dendsort(hclust(dist(moduleTraitCor)))
col_dend = dendsort(hclust(dist(t(moduleTraitCor))))

pdf(file = "Mcap2020/Figures/TagSeq/Module-trait-relationship-heatmap.pdf", height = 8, width = 8)
ht=Heatmap(moduleTraitCor, name = "Eigengene", row_title = "Gene Module", column_title = "Module-Lifestage Eigengene Correlation", 
        col = blueWhiteRed(50), 
        row_names_side = "left", 
        row_dend_side = "left",
        width = unit(5, "in"), 
        height = unit(4.5, "in"), 
        column_dend_reorder = TRUE, 
        #cluster_columns = hclust(dist(t(moduleTraitCor)), method = "average"),
        cluster_columns = col_dend,
        row_dend_reorder = TRUE,
        column_split = 6,
        row_split=4,
        column_dend_height = unit(.5, "in"),
        #cluster_rows = METree, 
        cluster_rows = row_dend, 
        row_gap = unit(2.5, "mm"), 
        border = TRUE,
        cell_fun = function(j, i, x, y, w, h, col) {
        if(heatmappval[i, j] < 0.05) {
            grid.text(sprintf("%s", heatmappval[i, j]), x, y, gp = gpar(fontsize = 10, fontface = "bold"))
        }
        else {
            grid.text(sprintf("%s", heatmappval[i, j]), x, y, gp = gpar(fontsize = 10, fontface = "plain"))
        }},
        column_names_gp =  gpar(fontsize = 12, border=FALSE),
        row_names_gp = gpar(fontsize = 12, alpha = 0.75, border = FALSE))
draw(ht)
dev.off()

```

# **Plot mean eigengene over developmental stages**  

View module eigengene data and make dataframe for Strader plots.  
```{r}
head(MEs)
names(MEs)
Strader_MEs <- MEs
Strader_MEs$lifestage <- metadata_ordered$lifestage
Strader_MEs$sample_id <- rownames(Strader_MEs)
head(Strader_MEs)

Strader_MEs<-Strader_MEs%>%
  droplevels() #drop unused level

dim(Strader_MEs)
head(Strader_MEs)


```

Plot mean module eigengene for each module.  

```{r}
#convert wide format to long format for plotting  
plot_MEs<-Strader_MEs%>%
  gather(., key="Module", value="Mean", 1:11)

dev.off() 

time_point_order = c("Egg (1 hpf)", "Embryo (5 hpf)", "Embryo (38 hpf)", "Embryo (65 hpf)", "Larvae (93 hpf)", "Larvae (163 hpf)", "Larvae (231 hpf)", "Metamorphosed Polyp (231 hpf)", "Attached Recruit (183 hpf)", "Attached Recruit (231 hpf)") #Set time_point order

expression_plots<-plot_MEs%>%
  group_by(Module, lifestage) %>%
  ggplot(aes(x=lifestage, y=Mean, group=lifestage)) +
  facet_grid(~Module)+
  geom_jitter(alpha = 0.5) +
  geom_boxplot(alpha=0) +
  scale_x_discrete(name="", limits=time_point_order) +
  #ylim(-0.5,1) +
  ylab("Mean Module Eigenegene") +
  geom_hline(yintercept = 0, linetype="dashed", color = "grey")+
  theme_bw() + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5, hjust=1, size = 12), #set x-axis label size
        axis.title.x=element_text(size = 14), #set x-axis title size
        axis.ticks.x=element_blank(), #No x-label ticks
        #axis.title.y=element_blank(), #No y-axis title
        axis.text.y=element_text(size = 14), #set y-axis label size, 
        panel.border = element_rect(color = "black", fill = NA, size = 1), #set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank(),
        plot.title = element_text(size=22)); expression_plots

ggsave(expression_plots, file="Mcap2020/Figures/TagSeq/expression_eigengene.pdf", height=6, width=20)
```

# **Calculating gene significance and module membership**  

We quantify associations of individual genes with life stage by defining Gene Significance GS as the absolute value of the correlation between the gene and the lifestage timepoint. For each module, we also define a quantitative measure of module membership (MM) as the correlation of the module eigengene and the gene expression profile.  

Define lifestage as a numeric value (ranging from 1 (egg, first timepoint)-10 (recruit, last time point)).  
```{r}
expressionProfile_data <- Strader_MEs

expressionProfile_data<-expressionProfile_data%>%
  droplevels() #drop unused level

cols.num <- c(1:8)

expressionProfile_data[cols.num] <- sapply(expressionProfile_data[cols.num],as.numeric)
sapply(expressionProfile_data, class)

dim(expressionProfile_data)
head(expressionProfile_data)

#need to get number of order of lifestage for this part 

expressionProfile_data$time_stage<-metadata$time_stage[match(expressionProfile_data$lifestage, metadata$lifestage)]
head(expressionProfile_data)

time_stage <- as.data.frame(expressionProfile_data$time_stage)
names(time_stage) = "lifestage_num"

time_stage$lifestage_num <- sub("_[^_]+$", "", time_stage$lifestage_num) #only extract the number to turn lifestage into numeric values 

time_stage$lifestage_num<-as.numeric(time_stage$lifestage_num)
dim(time_stage)
```

Generate data frame for module membership for each gene (absolute value of correlation between module eigengene and gene).  
- geneModuleMembership = correlation of the module eigengene and the gene 
- MMPvalue = p-value of correlation between module eigengene and the gene 
```{r}
names(datExpr) = gsub(pattern = "*.t1", replacement = "", x = names(datExpr))
#remove ".ti" from names in datExpr file and assign gene id as probe names

modNames = substring(names(MEs), 3) #pull out module names

geneModuleMembership = as.data.frame(cor(datExpr, MEs, use = "p"));
MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), nSamples));

names(geneModuleMembership) = paste("MM", modNames, sep="");
names(MMPvalue) = paste("p.MM", modNames, sep="");
```
 
Generate dataframe with Gene Significance GS as the absolute value of the correlation between the gene and the lifestage timepoint 
- geneTraitSignificance = correlation of the gene and lifestage time point 
- GSPvalue = p-value of correlation between gene and lifestage time point  
```{r}

geneTraitSignificance = as.data.frame(cor(datExpr, time_stage, use = "p"));
GSPvalue = as.data.frame(corPvalueStudent(as.matrix(geneTraitSignificance), nSamples));

names(geneTraitSignificance) = paste("GS.", names(time_stage), sep="");
names(GSPvalue) = paste("p.GS.", names(time_stage), sep="");
```

View correlation between module membership score and gene significance for each module: 

1, 2, 4, 6, 8, 11, 12, 14, 15, 16, 17, 18

```{r}
module = "1"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

module = "2"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

module = "4"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

module = "6"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

module = "8"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

module = "11"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

module = "12"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

module = "14"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

module = "15"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

module = "16"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

module = "17"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

module = "18"
column = match(module, modNames);
moduleGenes = moduleColors==module;

sizeGrWindow(7, 7);
par(mfrow = c(1,1));
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                   abs(geneTraitSignificance[moduleGenes, 1]),
                   xlab = paste("Module Membership in", module, "module"),
                   ylab = "Gene significance for body weight",
                   main = paste("Module membership vs. gene significance\n"),
                   cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col = module)

```

Generate a list of genes associated with each module. Change the module number in this code to look at respective module.   

```{r}
names(datExpr)[moduleColors=="14"]
```

# **Summary output of network analysis**  
## Make a data frame that connects genes, traits, and modules  

Import annotation file from Erin Chille repository.  
```{r}
Mcap.annot <- read_tsv("Mcap2020/Data/TagSeq/200824_Mcap_Blast_GO_KO.tsv", col_names = TRUE) #biological annotation information
head(Mcap.annot)
tail(Mcap.annot)
dim(Mcap.annot)
```

Match up genes in datExpr file to annotation file
```{r}
names(Mcap.annot)

probes = names(datExpr)

probes2annot = match(probes, Mcap.annot$gene_id) #match genes in datExpr to genes in annotation file

# The following is the number of probes without annotation... Should return 0.
sum(is.na(probes2annot))

row_nas<-which(is.na(probes2annot))

#view the genes that do not have a match in the annotation file
missing<-as.data.frame(probes[row_nas])
print(missing)
```
 
Create the starting data frame
```{r}
geneInfo0 = data.frame(gene_id = probes,
Accession = Mcap.annot$description[probes2annot],
Bitscore = Mcap.annot$bitscore[probes2annot],
eValue = Mcap.annot$eValue[probes2annot],
Description = Mcap.annot$protein_names[probes2annot],
KEGG = Mcap.annot$ko[probes2annot],
Annotation.GO.ID = Mcap.annot$GO_IDs[probes2annot],
Annotation.GO.Term = Mcap.annot$GO_terms[probes2annot],
moduleColor = moduleColors,
geneTraitSignificance,
GSPvalue)
```

Order modules by their significance for time_point
```{r}
modOrder = order(-abs(cor(MEs, time_stage, use = "p")))
```

Add module membership information in the chosen order
```{r}
for (mod in 1:ncol(geneModuleMembership))
{
oldNames = names(geneInfo0)
geneInfo0 = data.frame(geneInfo0, geneModuleMembership[, modOrder[mod]],
MMPvalue[, modOrder[mod]]);
names(geneInfo0) = c(oldNames, paste("MM.", modNames[modOrder[mod]], sep=""),
paste("p.MM.", modNames[modOrder[mod]], sep=""))
}
```

Order the genes in the geneInfo variable first by module color, then by geneTraitSignificance
```{r}
geneOrder = order(geneInfo0$moduleColor, -abs(geneInfo0$GS.lifestage_num));
geneInfo = geneInfo0[geneOrder, ]
head(geneInfo)
```

Save geneInfo as a CSV
```{r}
head(geneInfo)
geneInfo$Annotation.GO.ID <- gsub(";NA", "", geneInfo$Annotation.GO.ID) #Remove NAs
geneInfo$Annotation.GO.ID <- gsub("NA", "", geneInfo$Annotation.GO.ID) #Remove NAs

write.csv(geneInfo, file = "Mcap2020/Output/TagSeq/geneInfo.csv") #gene info for reference/supplement

```

# **Other approaches for network visualization**  

Use TOM to visualize network. This can take a very long time to run, remove # if plot is desired.  
```{r}

# Transform dissTOM with a power to make moderately strong connections more visible in the heatmap
#plotTOM = dissTOM^7;
# Set diagonal to NA for a nicer plot
#diag(plotTOM) = NA;
# Call the plot function
#dev.off()
#sizeGrWindow(9,9)
#TOMplot(plotTOM, geneTree, moduleColors, main = "Network heatmap plot, all genes")

```

Plot with a reduced set of genes.  

```{r}
# nGenes = ncol(datExpr)
# nSamples = nrow(datExpr)
# 
# nSelect = 400
# # For reproducibility, we set the random seed
# set.seed(10);
# select = sample(nGenes, size = nSelect);
# selectTOM = dissTOM[select, select];
# # There's no simple way of restricting a clustering tree to a subset of genes, so we must re-cluster.
# selectTree = hclust(as.dist(selectTOM), method = "average")
# selectColors = moduleColors[select];
# # Open a graphical window
# sizeGrWindow(9,9)
# # Taking the dissimilarity to a power, say 10, makes the plot more informative by effectively changing 
# # the color palette; setting the diagonal to NA also improves the clarity of the plot
# plotDiss = selectTOM^7;
# diag(plotDiss) = NA;
# TOMplot(plotDiss, selectTree, selectColors, main = "Network heatmap plot, selected genes")
```

Other plots from tutorial that are not used, but here for reference.  
```{r}
# Recalculate module eigengenes
# MEs = moduleEigengenes(datExpr, moduleColors)$eigengenes
# # Isolate weight from the clinical traits
# lifestage = as.data.frame(lifestage$lifestage_num);
# names(lifestage) = "lifestage"
# # Add the weight to existing module eigengenes
# MET = orderMEs(cbind(MEs, lifestage))
```

```{r}
# Plot the dendrogram
# sizeGrWindow(6,6);
# par(cex = 1.0)
# plotEigengeneNetworks(MET, "Eigengene dendrogram", marDendro = c(0,4,2,0),
#                       plotHeatmaps = FALSE)
# # Plot the heatmap matrix (note: this plot will overwrite the dendrogram plot)
# par(cex = 1.0)
# plotEigengeneNetworks(MET, "Eigengene adjacency heatmap", marHeatmap = c(3,4,2,2),
#                       plotDendrograms = FALSE, xLabelsAngle = 90)
```


# **Output files for functional enrichment** 

GOMWU: https://github.com/z0on/GO_MWU  

Make output tables for each module with the following:  

1. "table of GO annotations for your sequences: two-column (gene id - GO terms), tab-delimited, one line per gene, multiple GO terms separated by semicolon. If you have multiple lines per gene, use nrify_GOtable.pl to merge them."

```{r}
#generate a table of GO annotations for all sequences   
go_mwu_terms<-geneInfo%>%
  select(gene_id, Annotation.GO.ID, Annotation.GO.Term)%>%
  write.table(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_terms.tab", row.names=FALSE, quote=FALSE, sep = "\t", col.names=FALSE)
```

2. "table of measure of interest for your sequences: two columns of comma-separated values: gene id, continuous measure of change such as log(fold-change). To perform standard GO enrichment analysis based on Fisher's exact test, use binary measure (1 or 0, i.e., either sgnificant or not). To analyze modules derived from WGCNA, specify 0 for genes not included in the module and the kME value (number between 0 and 1, module membership score) for genes included in the module."

Subset genes that are included in each module, substitute 0 for module membership scores for genes included in module. 

Module list: 1, 2, 4, 6, 8, 11, 12, 14, 15, 16, 17, 18     
```{r}
go_mwu_me1<-geneInfo%>%
  select(gene_id, MM.1, moduleColor)%>%
  mutate(MM.1 = if_else(moduleColor==1, 0, MM.1))%>% #if a gene is associated with a module (p<0.05), keep the module membership score. If not, replace MM with a 0 per GOMWU instructions. 
  select(gene_id, MM.1)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me1.csv", row.names=FALSE, quote = FALSE)

go_mwu_me2<-geneInfo%>%
  select(gene_id, MM.2, moduleColor)%>%
  mutate(MM.2 = if_else(moduleColor==2, 0, MM.2))%>%
  select(gene_id, MM.2)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me2.csv", row.names=FALSE, quote = FALSE)
  
go_mwu_me4<-geneInfo%>%
  select(gene_id, MM.4, moduleColor)%>%
  mutate(MM.4 = if_else(moduleColor==4, 0, MM.4))%>%
  select(gene_id, MM.4)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me4.csv", row.names=FALSE, quote = FALSE)

go_mwu_me6<-geneInfo%>%
  select(gene_id, MM.6, moduleColor)%>%
  mutate(MM.6 = if_else(moduleColor==6, 0, MM.6))%>%
  select(gene_id, MM.6)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me6.csv", row.names=FALSE, quote = FALSE)

go_mwu_me8<-geneInfo%>%
  select(gene_id, MM.8, moduleColor)%>%
  mutate(MM.8 = if_else(moduleColor==8, 0, MM.8))%>%
  select(gene_id, MM.8)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me8.csv", row.names=FALSE, quote = FALSE)
  
go_mwu_me11<-geneInfo%>%
  select(gene_id, MM.11, moduleColor)%>%
  mutate(MM.11 = if_else(moduleColor==11, 0, MM.11))%>%
  select(gene_id, MM.11)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me11.csv", row.names=FALSE, quote = FALSE)
  
go_mwu_me12<-geneInfo%>%
  select(gene_id, MM.12, moduleColor)%>%
  mutate(MM.12 = if_else(moduleColor==12, 0, MM.12))%>%
  select(gene_id, MM.12)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me12.csv", row.names=FALSE, quote = FALSE)

go_mwu_me14<-geneInfo%>%
  select(gene_id, MM.14, moduleColor)%>%
  mutate(MM.14 = if_else(moduleColor==14, 0, MM.14))%>%
  select(gene_id, MM.14)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me14.csv", row.names=FALSE, quote = FALSE)

go_mwu_me15<-geneInfo%>%
  select(gene_id, MM.15, moduleColor)%>%
  mutate(MM.15 = if_else(moduleColor==15, 0, MM.15))%>%
  select(gene_id, MM.15)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me15.csv", row.names=FALSE, quote = FALSE)

go_mwu_me14<-geneInfo%>%
  select(gene_id, MM.14, moduleColor)%>%
  mutate(MM.14 = if_else(moduleColor==14, 0, MM.14))%>%
  select(gene_id, MM.14)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me14.csv", row.names=FALSE, quote = FALSE)

go_mwu_me16<-geneInfo%>%
  select(gene_id, MM.16, moduleColor)%>%
  mutate(MM.16 = if_else(moduleColor==16, 0, MM.16))%>%
  select(gene_id, MM.16)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me16.csv", row.names=FALSE, quote = FALSE)

go_mwu_me17<-geneInfo%>%
  select(gene_id, MM.17, moduleColor)%>%
  mutate(MM.17 = if_else(moduleColor==17, 0, MM.17))%>%
  select(gene_id, MM.17)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me17.csv", row.names=FALSE, quote = FALSE)

go_mwu_me18<-geneInfo%>%
  select(gene_id, MM.18, moduleColor)%>%
  mutate(MM.18 = if_else(moduleColor==18, 0, MM.18))%>%
  select(gene_id, MM.18)%>%
  write.csv(., "Mcap2020/Output/TagSeq/GOMWU/go_mwu_me18.csv", row.names=FALSE, quote = FALSE)
```

Next Run GOMWU script in TagSeq Output folder.  

# **Tracking expression of specific genes**  

## Plot function specific genes of interest  

Format our gene vst data frame for plotting.  
```{r}
gene_expr<-as.matrix(datExpr)
gene_expr<-t(gene_expr)
gene_expr<-as.data.frame(gene_expr, row.names = rownames(gene_expr), make.names=TRUE)
```

Output for other analysis
```{r}
write.csv(gene_expr, "Mcap2020/Output/TagSeq/transformed_filtered_genes.csv")
```

Next, from our Gene Info data frame, extract a list of gene names with specific functions of interest.  

### Create lists of genes  

Read in required data frames.  
```{r}
geneInfo<- read.csv("Mcap2020/Output/TagSeq/geneInfo.csv") #gene info for reference/supplement
head(geneInfo)

gene_expr<-read.csv("Mcap2020/Output/TagSeq/transformed_filtered_genes.csv")
rownames(gene_expr)<-gene_expr$X
gene_expr<-gene_expr%>%
  select(!X)
```

1. Create list of expression of genes for carbonic anhydrase: 

```{r}
list_ca<-geneInfo%>%
  filter(grepl('carbonate dehydratase activity', Annotation.GO.Term))

head(list_ca)

carbanhyd<-list_ca$gene_id

#carbanhyd<-c("adi2mcaRNA26720_R2", "adi2mcaRNA26720_R3")

carbanhyd_genes<-gene_expr[rownames(gene_expr) %in% carbanhyd, ]
carbanhyd_genes

carbanhyd_genes$Function<-c("Carbonic Anhydrase")
carbanhyd_genes$gene_id<-rownames(carbanhyd_genes)
rownames(carbanhyd_genes) <- NULL
```

2. Create list of expression of genes for glycogen synthase: 

```{r}
list_gs<-geneInfo%>%
  filter(grepl('glycogen biosynthetic process', Annotation.GO.Term))

head(list_gs)

glycsyn<-list_gs$gene_id

#glycsyn<-c("g16014", "g19866")
glycsyn_genes<-gene_expr[rownames(gene_expr) %in% glycsyn, ]
glycsyn_genes

glycsyn_genes$Function<-c("Glycogen Synthase")
glycsyn_genes$gene_id<-rownames(glycsyn_genes)
rownames(glycsyn_genes) <- NULL
```

3. Create list of expression of genes for ammonium transporters: 

```{r}
list_at<-geneInfo%>%
  filter(grepl('ammonium transmembrane transporter', Annotation.GO.Term))

head(list_at)

ammontr<-list_at$gene_id

#ammontr<-c("adi2mcaRNA16885_R0", "adi2mcaRNA8162_R0", "g13023", "g15870", "g16203", "g21992", "g26425")
ammontr_genes<-gene_expr[rownames(gene_expr) %in% ammontr, ]
ammontr_genes

ammontr_genes$Function<-c("Ammonium Transporter")
ammontr_genes$gene_id<-rownames(ammontr_genes)
rownames(ammontr_genes) <- NULL
```

4. Create list of expression of genes for glutathione peroxidase activity: 

```{r}
list_gp<-geneInfo%>%
  filter(grepl('glutathione peroxidase', Annotation.GO.Term))

head(list_gp)

glutperox<-list_gp$gene_id

#glutperox<-c("g26468", "g30035", "g34381", "g50455")
glutperox_genes<-gene_expr[rownames(gene_expr) %in% glutperox, ]
glutperox_genes

glutperox_genes$Function<-c("Glutathione Peroxidase")
glutperox_genes$gene_id<-rownames(glutperox_genes)
rownames(glutperox_genes) <- NULL
```

5. Create list of expression of genes for glucose transporters: 

```{r}
list_gt<-geneInfo%>%
  filter(grepl('glucose transmembrane transporter', Annotation.GO.Term))

head(list_gt)

gluctr<-list_gt$gene_id

#gluctr<-c("g65554", "g27005")
gluctr_genes<-gene_expr[rownames(gene_expr) %in% gluctr, ]
gluctr_genes

gluctr_genes$Function<-c("Glucose Transporter")
gluctr_genes$gene_id<-rownames(gluctr_genes)
rownames(gluctr_genes) <- NULL
```

6. Create list of expression of genes for H+ATPase: 

```{r}
list_atp<-geneInfo%>%
  filter(grepl('ATPase-coupled transmembrane transporter activity', Annotation.GO.Term))

head(list_atp)

atpase<-list_atp$gene_id

#atpase<-c("g36721", "g42331", "g58299")
atpase_genes<-gene_expr[rownames(gene_expr) %in% atpase, ]
atpase_genes

atpase_genes$Function<-c("H+ ATPase")
atpase_genes$gene_id<-rownames(atpase_genes)
rownames(atpase_genes) <- NULL
```

7. Create list of expression of genes for glucose homeostasis: 

```{r}
list_gh<-geneInfo%>%
  filter(grepl('glucose homeostasis', Annotation.GO.Term))

head(list_gh)

glucosehomeo<-list_gh$gene_id

#glucosehomeo<-c("g16598", "g64416")
glucosehomeo_genes<-gene_expr[rownames(gene_expr) %in% glucosehomeo, ]
glucosehomeo_genes

glucosehomeo_genes$Function<-c("Glucose Homeostasis")
glucosehomeo_genes$gene_id<-rownames(glucosehomeo_genes)
rownames(glucosehomeo_genes) <- NULL
```

8. Create list of expression of genes for apoptosis regulation: 

```{r}
list_ar<-geneInfo%>%
  filter(grepl('regulation of apoptotic process', Annotation.GO.Term))

head(list_ar)

apoptosis<-list_ar$gene_id

#apoptosis<-c("g7993", "g25930", "g62182", "g25928")
apoptosis_genes<-gene_expr[rownames(gene_expr) %in% apoptosis, ]
apoptosis_genes

apoptosis_genes$Function<-c("Apoptosis Regulation")
apoptosis_genes$gene_id<-rownames(apoptosis_genes)
rownames(apoptosis_genes) <- NULL
```

9. Create list of expression of genes for cellular communication/signaling: 

```{r}
list_cc<-geneInfo%>%
  filter(grepl('cell-cell signaling', Annotation.GO.Term))

head(list_cc)

communication<-list_cc$gene_id

#communication<-c("g47126", "g32455", "g29355")

communication_genes<-gene_expr[rownames(gene_expr) %in% communication, ]
communication_genes

communication_genes$Function<-c("Cell Communication")
communication_genes$gene_id<-rownames(communication_genes)
rownames(communication_genes) <- NULL
```

10. Create list of expression of genes for glucose metabolism: 

```{r}
list_cm<-geneInfo%>%
  filter(grepl('glucose metabolic process', Annotation.GO.Term))

head(list_cm)

carbs<-list_cm$gene_id

#carbs<-c("g27530", "g27612", "g48204", "g36198", "g15393", "g27646")

carbs_genes<-gene_expr[rownames(gene_expr) %in% carbs, ]
carbs_genes

carbs_genes$Function<-c("Glucose Metabolism")
carbs_genes$gene_id<-rownames(carbs_genes)
rownames(carbs_genes) <- NULL
```

11. Create list of expression of genes for fatty acid metabolism: 

```{r}
list_fa<-geneInfo%>%
  filter(grepl('fatty acid metabolic process', Annotation.GO.Term))

head(list_fa)

fattyacid<-list_fa$gene_id

fattyacid_genes<-gene_expr[rownames(gene_expr) %in% fattyacid, ]
fattyacid_genes

fattyacid_genes$Function<-c("Fatty Acid Metabolism")
fattyacid_genes$gene_id<-rownames(fattyacid_genes)
rownames(fattyacid_genes) <- NULL
```

12. Create list of expression of genes for all carbohydrate metabolism: 

```{r}
list_carbs<-geneInfo%>%
  filter(grepl('carbohydrate metabolic process', Annotation.GO.Term))

head(list_carbs)

allcarbs<-list_carbs$gene_id

allcarbs_genes<-gene_expr[rownames(gene_expr) %in% allcarbs, ]
allcarbs_genes

allcarbs_genes$Function<-c("Carbohydrate Metabolism")
allcarbs_genes$gene_id<-rownames(allcarbs_genes)
rownames(allcarbs_genes) <- NULL
```

13. Create list of expression of genes for lipid metabolism: 

```{r}
list_lipids<-geneInfo%>%
  filter(grepl('lipid metabolic process', Annotation.GO.Term))

head(list_lipids)

lipids<-list_lipids$gene_id

lipids_genes<-gene_expr[rownames(gene_expr) %in% lipids, ]
lipids_genes

lipids_genes$Function<-c("Lipid Metabolism")
lipids_genes$gene_id<-rownames(lipids_genes)
rownames(lipids_genes) <- NULL
```

14. Create list of expression of genes for glutamine synthase: 

```{r}
list_glut<-geneInfo%>%
  filter(grepl('glutamate-ammonia ligase', Annotation.GO.Term))

head(list_glut)

glutamine<-list_glut$gene_id

glutamine_genes<-gene_expr[rownames(gene_expr) %in% glutamine, ]
glutamine_genes

glutamine_genes$Function<-c("Glutamine Synthase")
glutamine_genes$gene_id<-rownames(glutamine_genes)
rownames(glutamine_genes) <- NULL
```
Not in expressed genes frame.

15. Create list of expression of genes for methyltransferase: 

```{r}
list_methyl<-geneInfo%>%
  filter(grepl('methyltransferase', Annotation.GO.Term))

head(list_methyl)

methyl_genes<-list_methyl$gene_id

methyl_genes<-gene_expr[rownames(gene_expr) %in% methyl_genes, ]
methyl_genes

methyl_genes$Function<-c("Methyltransferases")
methyl_genes$gene_id<-rownames(methyl_genes)
rownames(methyl_genes) <- NULL
```

```{r}
specific_genes<-rbind(carbanhyd_genes, glycsyn_genes, ammontr_genes, glutperox_genes, atpase_genes, glucosehomeo_genes, apoptosis_genes, communication_genes, carbs_genes, fattyacid_genes, allcarbs_genes, gluctr_genes, lipids_genes, methyl_genes)
```

Gather the data set and merge in metadata information.  

```{r}
specific_genes<-specific_genes%>%
  gather(key="Sample", value="vst", 1:38)

#bring in developmental metadata 
keys<-read.csv("Mcap2020/Data/lifestage_metadata.csv")

specific_genes$code<-metadata_ordered$code[match(specific_genes$Sample, metadata_ordered$sample_id)]

specific_genes$lifestage<-keys$lifestage[match(specific_genes$code, keys$code)]

specific_genes$group<-keys$group[match(specific_genes$code, keys$code)]
head(specific_genes)
```

Plot the data for each function.  

Reorder levels for development.    
```{r}
group_order<-c("Egg", "Embryo", "Larvae", "Metamorphosed Polyp", "Attached Recruit")

specific_genes$group<-factor(specific_genes$group, levels = group_order)

levels(as.factor(specific_genes$lifestage))

lifestage_order<-c("Egg (1 hpf)", "Embryo (5 hpf)", "Embryo (38 hpf)", "Embryo (65 hpf)", "Larvae (93 hpf)", "Larvae (163 hpf)", "Larvae (231 hpf)", "Metamorphosed Polyp (231 hpf)", "Attached Recruit (183 hpf)", "Attached Recruit (231 hpf)")

specific_genes$lifestage<-factor(specific_genes$lifestage, levels = lifestage_order)

levels(as.factor(specific_genes$lifestage))
```

### Plot each function  

Plot H+ATPase
```{r}
#show mean
atpase_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="H+ ATPase")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5.5,6)+
    theme_classic() + 
    geom_text(x=1.7, y=5.5, label="N=54 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); atpase_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="H+ ATPase")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot apoptosis regulation  
```{r}
#show mean
apoptosis_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Apoptosis Regulation")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5.5,6)+
    theme_classic() + 
    geom_text(x=1.7, y=5.5, label="N=103 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); apoptosis_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Apoptosis Regulation")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot Carbonic Anhydrase

```{r}
anhydrase_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Carbonic Anhydrase")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5,7)+
    theme_classic() + 
    geom_text(x=1.7, y=5, label="N=17 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); anhydrase_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Carbonic Anhydrase")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot Ammonium Transporter

```{r}
ammonium_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Ammonium Transporter")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5,7)+
    theme_classic() +
    geom_text(x=1.7, y=5, label="N=5 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); ammonium_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Ammonium Transporter")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot Glucose Homeostasis

```{r}
glucose_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Glucose Homeostasis")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5.25,6.5)+
    theme_classic() + 
    geom_text(x=1.7, y=5.25, label="N=3 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); glucose_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Glucose Homeostasis")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot Glutathione Peroxidase

```{r}
glutathione_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Glutathione Peroxidase")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
    geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5,8)+
    theme_classic() + 
    geom_text(x=1.7, y=5, label="N=3 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); glutathione_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Glutathione Peroxidase")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot Glycogen Synthase

```{r}
glycogen_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Glycogen Synthase")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5.25,6.25)+
    theme_classic() + 
    geom_text(x=1.7, y=5.25, label="N=5 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); glycogen_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Glycogen Synthase")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot Cell Communication
```{r}
communication_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Cell Communication")%>%
    
  ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5,6)+
    theme_classic() + 
    geom_text(x=1.7, y=5, label="N=6 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); communication_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Cell Communication")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot Glucose Metabolism
```{r}
carb_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Glucose Metabolism")%>%
    
  ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5,6.25)+
    theme_classic() + 
    geom_text(x=1.7, y=5, label="N=3 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); carb_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Glucose Metabolism")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot Fatty Acid Metabolism
```{r}
fatty_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Fatty Acid Metabolism")%>%
    
  ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5,6.25)+
    theme_classic() + 
    geom_text(x=1.7, y=5, label="N=6 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); fatty_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Fatty Acid Metabolism")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot Carbohydrate Metabolism
```{r}
allcarbs_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Carbohydrate Metabolism")%>%
    
  ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5,6.25)+
    theme_classic() + 
    geom_text(x=1.7, y=5, label="N=95 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); allcarbs_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Carbohydrate Metabolism")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot Glucose Transporters
```{r}
gluctrans_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Glucose Transporter")%>%
    
  ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5,7.5)+
    theme_classic() + 
    geom_text(x=1.7, y=5, label="N=95 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); gluctrans_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Glucose Transporter")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot Lipid Metabolism  
```{r}
lipid_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Lipid Metabolism")%>%
    
  ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5,7)+
    theme_classic() + 
    geom_text(x=1.7, y=5, label="N=68 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); lipid_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Lipid Metabolism")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

Plot methylation 
```{r}
methyl_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Methyltransferases")%>%
    
  ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), color="black", position = position_dodge(0.1)) +
    geom_point(size=2, position = position_dodge(0.1)) + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", position=position_dodge(0.1), size=0.5) + 
    xlab("Lifestage") + 
    ylab("VST expression")+
    ylim(5.5,6.5)+
    theme_classic() + 
    geom_text(x=1.7, y=5.5, label="N=104 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
            axis.text.x=element_text(angle=90),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12)
      ); methyl_plot
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Methyltransferases")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

### Assemble all plots together.  
```{r}
library(cowplot)

carb_plot2<-carb_plot+theme(axis.text.x=element_blank(), axis.title.x=element_blank())
allcarbs_plot2<-allcarbs_plot+theme(axis.text.x=element_blank(), axis.title.x=element_blank())
anhydrase_plot2<-anhydrase_plot+theme(axis.text.x=element_blank(), axis.title.x=element_blank())
communication_plot2<-communication_plot+theme(axis.text.x=element_blank(), axis.title.x=element_blank())
fatty_plot2<-fatty_plot+theme(axis.text.x=element_blank(), axis.title.x=element_blank())
gluctrans_plot2<-gluctrans_plot+theme(axis.text.x=element_blank(), axis.title.x=element_blank())
ammonium_plot2<-ammonium_plot+theme(axis.text.x=element_blank(), axis.title.x=element_blank())
lipid_plot2<-lipid_plot+theme(axis.text.x=element_blank(), axis.title.x=element_blank())

gene_plots<-plot_grid(allcarbs_plot2, carb_plot2, fatty_plot2, gluctrans_plot2, lipid_plot2, anhydrase_plot, communication_plot, ammonium_plot, atpase_plot,  apoptosis_plot, apoptosis_plot, gluctrans_plot, ncol=5, nrow=2, rel_heights = c(0.5,1))
                      
ggsave(filename="Mcap2020/Figures/TagSeq/gene_expression_class.jpg", plot=gene_plots, dpi=300, width=22, height=9, units="in")
```

