---
title: "Gene expression indicators"
author: "Ariana S Huffmyer"
date: "2023"
output: html_document
editor_options: 
  chunk_output_type: console
---

Plotting gene expression of genes in categories of interest.  

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# The following setting is important, do not omit.
options(stringsAsFactors = FALSE) #Set Strings to character
```

Load required libraries.  

```{r}
if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse') 
if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer') 
if ("ggplot2" %in% rownames(installed.packages()) == 'FALSE') install.packages('ggplot2') 
if ("cowplot" %in% rownames(installed.packages()) == 'FALSE') install.packages('cowplot') 

library("tidyverse")
library("RColorBrewer")
library("ggplot2")
library("cowplot")

```

# Load data frames 

Load the gene information dataframe with metadata for each gene and the filtered and transformed gene count matrix from scripts in step 1.  

```{r}
geneInfo<-read_csv("Mcap2020/Output/TagSeq/GenomeV3/geneInfo.csv")
head(geneInfo)
  
gene_expr<-read_csv("Mcap2020/Output/TagSeq/GenomeV3/transformed_filtered_gene_counts.csv")
head(gene_expr)
colnames(gene_expr)[1] <- "gene_id"

goseq_increasing<-read_csv("Mcap2020/Output/TagSeq/GenomeV3/GoSeq/goseq_pattern_Increasing.csv")

goseq_decreasing<-read_csv("Mcap2020/Output/TagSeq/GenomeV3/GoSeq/goseq_pattern_Decreasing.csv")

goseq_mid<-read_csv("Mcap2020/Output/TagSeq/GenomeV3/GoSeq/goseq_pattern_Mid.csv")

head(goseq_mid)
```

Bind GO term files together from each ontogenetic pattern into a list. We will use this to pull out gene ID's associated with these GO terms.  
```{r}
goterms<-rbind(goseq_increasing,goseq_decreasing,goseq_mid)

goterms<-goterms%>%
  dplyr::select(GOterm, term, ontology)%>%
  unique()
```

# Extract lists of genes of interest 

Next, extract a list of GO terms that identify genes of interest in the following categories. This will then be used to select gene ID's that have these GO terms and plot their expression.   

## Carbohydrate metabolism 

Create list of expression of genes for carbohydrate general metabolism: 

```{r}
carb_terms<-c("carbohydrate derivative metabolic process", "carbohydrate metabolic process", "carbohydrate homeostasis", "regulation of carbohydrate metabolic process", "carbohydrate catabolic process")
              
list_carbs<-goterms%>%
    filter(grepl(paste(carb_terms, collapse = "|"), term))%>%
    dplyr::select(GOterm)

#pull out any genes that have a GO term in the above list 

genes_carbs<-geneInfo%>%
  filter(grepl(paste(list_carbs$GOterm, collapse = "|"), Annotation.GO.ID))%>%
  dplyr::select(gene_id)

data_carbs<-gene_expr[gene_expr$gene_id %in% genes_carbs$gene_id, ]
data_carbs

data_carbs$Function<-c("Carbohydrate Metabolism")
```
There are 725 genes that match carbohydrate metabolism key words. 

## Glucose metabolism 

Create list of expression of genes for glucose metabolism: 

```{r}
glucose_terms<-c("glucose homeostasis", "response to glucose", "glucose metabolic process", "regulation of glucose metabolic process", "glucose catabolic process")
              
list_glucose<-goterms%>%
    filter(grepl(paste(glucose_terms, collapse = "|"), term))%>%
    dplyr::select(GOterm)

#pull out any genes that have a GO term in the above list 
genes_glucose<-geneInfo%>%
  filter(grepl(paste(list_glucose$GOterm, collapse = "|"), Annotation.GO.ID))%>%
  dplyr::select(gene_id)

data_glucose<-gene_expr[gene_expr$gene_id %in% genes_glucose$gene_id, ]
data_glucose

data_glucose$Function<-c("Glucose Metabolism")
```
There are 232 genes that match glucose metabolism key words. 

## Glucose transport 

Create list of expression of genes for glucose transport: 

```{r}
glucose_transport_terms<-c("regulation of glucose import", "regulation of glucose transmembrane transport")
              
list_glucose_transport<-goterms%>%
    filter(grepl(paste(glucose_transport_terms, collapse = "|"), term))%>%
    dplyr::select(GOterm)

#pull out any genes that have a GO term in the above list 
genes_glucose_transport<-geneInfo%>%
  filter(grepl(paste(list_glucose_transport$GOterm, collapse = "|"), Annotation.GO.ID))%>%
  dplyr::select(gene_id)

data_glucose_transport<-gene_expr[gene_expr$gene_id %in% genes_glucose_transport$gene_id, ]
data_glucose_transport

data_glucose_transport$Function<-c("Glucose Transport")
```
There are 44 genes that match glucose transport metabolism key words.

## Lipid metabolism 

Create list of expression of genes for lipid metabolism: 

```{r}
lipid_terms<-c("lipid metabolic process", "regulation of lipid metabolic process", "lipid oxidation")
              
list_lipids<-goterms%>%
    filter(grepl(paste(lipid_terms, collapse = "|"), term))%>%
    dplyr::select(GOterm)

#pull out any genes that have a GO term in the above list 
genes_lipids<-geneInfo%>%
  filter(grepl(paste(list_lipids$GOterm, collapse = "|"), Annotation.GO.ID))%>%
  dplyr::select(gene_id)

data_lipids<-gene_expr[gene_expr$gene_id %in% genes_lipids$gene_id, ]
data_lipids

data_lipids$Function<-c("Lipid Metabolism")
```
There are 519 genes that match lipid metabolism key words.

## Fatty acid metabolism 

Create list of expression of genes for fatty acid metabolism: 

```{r}
fattyacid_terms<-c("fatty acid metabolic process", "fatty acid oxidation", "fatty acid beta-oxidation", "long-chain fatty acid metabolic process", "unsaturated fatty acid metabolic process")
              
list_fattyacid<-goterms%>%
    filter(grepl(paste(fattyacid_terms, collapse = "|"), term))%>%
    dplyr::select(GOterm)

#pull out any genes that have a GO term in the above list 
genes_fattyacid<-geneInfo%>%
  filter(grepl(paste(list_fattyacid$GOterm, collapse = "|"), Annotation.GO.ID))%>%
  dplyr::select(gene_id)

data_fattyacid<-gene_expr[gene_expr$gene_id %in% genes_fattyacid$gene_id, ]
data_fattyacid

data_fattyacid$Function<-c("Fatty Acid Metabolism")
```
There are 159 genes that match fatty acid metabolism key words. 

## Ammonium metabolism 

Create list of expression of genes for ammonium metabolism: 

```{r}
nitrogen_terms<-c("ammonium ion metabolic process", "response to ammonium ion")
              
list_nitrogen<-goterms%>%
    filter(grepl(paste(nitrogen_terms, collapse = "|"), term))%>%
    dplyr::select(GOterm)

#pull out any genes that have a GO term in the above list 
genes_nitrogen<-geneInfo%>%
  filter(grepl(paste(list_nitrogen$GOterm, collapse = "|"), Annotation.GO.ID))%>%
  dplyr::select(gene_id)

data_nitrogen<-gene_expr[gene_expr$gene_id %in% genes_nitrogen$gene_id, ]
data_nitrogen

data_nitrogen$Function<-c("Ammonium Metabolism")
```
There are 142 genes that match ammonium metabolism key words. 

## Oxidative stress genes

Create list of expression of genes for symbiosis function: 

```{r}
oxidative_terms<-c("antioxidant activity", "response to oxidative stress", "peroxidase activity")
              
list_oxidative<-goterms%>%
    filter(grepl(paste(oxidative_terms, collapse = "|"), term))%>%
    dplyr::select(GOterm)

#pull out any genes that have a GO term in the above list 
genes_oxidative<-geneInfo%>%
  filter(grepl(paste(list_oxidative$GOterm, collapse = "|"), Annotation.GO.ID))%>%
  dplyr::select(gene_id)

data_oxidative<-gene_expr[gene_expr$gene_id %in% genes_oxidative$gene_id, ]
data_oxidative

data_oxidative$Function<-c("Oxidative Stress")
```
There are 295 genes that match oxidative stress key words. 

## Bicarbonate genes

Create list of expression of genes for bicarbonate transport: 

```{r}
bicarb_terms<-c("bicarbonate transport")
              
list_bicarb<-goterms%>%
    filter(grepl(paste(bicarb_terms, collapse = "|"), term))%>%
    dplyr::select(GOterm)

#pull out any genes that have a GO term in the above list 
genes_bicarb<-geneInfo%>%
  filter(grepl(paste(list_bicarb$GOterm, collapse = "|"), Annotation.GO.ID))%>%
  dplyr::select(gene_id)

data_bicarb<-gene_expr[gene_expr$gene_id %in% genes_bicarb$gene_id, ]
data_bicarb

data_bicarb$Function<-c("Bicarbonate Transport")
```
There are 17 genes that match bicarbonate transport key words. 

## Merge all data frames together  

```{r}
# Find data frames that start with "genes_"
matching_df_names <- ls(pattern = "^data_")

# Initialize an empty dataframe to store the combined result
specific_genes <- data.frame()

# Bind the matching data frames together row-wise
for (df_name in matching_df_names) {
  df <- get(df_name)  # Access the data frame
  specific_genes <- rbind(specific_genes, df)  # Bind the data frame
}
```

# Gather metadata 

Gather the data set and merge in metadata information.  

```{r}
specific_genes<-specific_genes%>%
  gather(key="Sample", value="vst", 2:39)

#bring in developmental metadata 
metadata<-read.csv("Mcap2020/Data/TagSeq/metadata_ordered.csv")
keys<-read.csv("Mcap2020/Data/lifestage_metadata.csv")

#add lifestage information
specific_genes$lifestage<-metadata$lifestage[match(specific_genes$Sample, metadata$sample_id)]
specific_genes$group<-keys$group[match(specific_genes$lifestage, keys$lifestage)]

head(specific_genes)
```

Reorder levels for development.    
```{r}
group_order<-c("Egg", "Embryo", "Larvae", "Metamorphosed Polyp", "Attached Recruit")

specific_genes$group<-factor(specific_genes$group, levels = group_order)

levels(as.factor(specific_genes$lifestage))

lifestage_order<-c("Egg (1 hpf)", "Embryo (5 hpf)", "Embryo (38 hpf)", "Embryo (65 hpf)", "Larvae (93 hpf)", "Larvae (163 hpf)", "Larvae (231 hpf)", "Metamorphosed Polyp (231 hpf)", "Attached Recruit (183 hpf)", "Attached Recruit (231 hpf)")

specific_genes$lifestage<-factor(specific_genes$lifestage, levels = lifestage_order)

levels(as.factor(specific_genes$lifestage))
```

# Plot expression for each function category 

## Carbohydrate metabolism  

Run a one way anova to test variation in expression across life stages. 
```{r}
specific_genes%>%
  filter(Function=="Carbohydrate Metabolism")%>%
  
  aov(vst~lifestage, data=.)%>%
  summary()
```

```{r}
carb_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Carbohydrate Metabolism")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), size=1, color="black") +
    geom_point(size=4, colour="black", fill="white") + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", size=0.5) + 
    xlab("") + 
    ylab("Gene expression (vst)")+
    ylim(5.9,6.7)+
    theme_classic() + 
    geom_text(x=2, y=6.7, label="p<0.001", color="black", size=4)+
    geom_text(x=1.5, y=5.9, label="N=725 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      axis.text.x=element_text(angle=45, hjust=1, color="black"),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12), 
      strip.text=element_text(size=12, face="bold")
      ); carb_plot

ggsave(carb_plot, file="Mcap2020/Figures/TagSeq/GenomeV3/SpecificGenes/carb_metabolism.png", width=6, height=6)
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Carbohydrate Metabolism")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

## Glucose Metabolism 

Run a one way anova to test variation in expression across life stages. 
```{r}
specific_genes%>%
  filter(Function=="Glucose Metabolism")%>%
  
  aov(vst~lifestage, data=.)%>%
  summary()
```

```{r}
glucose_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Glucose Metabolism")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), size=1, color="black") +
    geom_point(size=4, colour="black", fill="white") + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", size=0.5) + 
    xlab("") + 
    ylab("Gene expression (vst)")+
    geom_text(x=2, y=6.7, label="p<0.001", color="black", size=4)+
    ylim(5.9,6.7)+
    theme_classic() + 
    geom_text(x=1.5, y=5.9, label="N=232 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      axis.text.x=element_text(angle=45, hjust=1, color="black"),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12), 
      strip.text=element_text(size=12, face="bold")
      ); glucose_plot

ggsave(glucose_plot, file="Mcap2020/Figures/TagSeq/GenomeV3/SpecificGenes/glucose_metabolism.png", width=6, height=6)
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Glucose Metabolism")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

## Glucose Transport 

Run a one way anova to test variation in expression across life stages. 
```{r}
specific_genes%>%
  filter(Function=="Glucose Transport")%>%
  
  aov(vst~lifestage, data=.)%>%
  summary()
```

```{r}
glucose_transport_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Glucose Transport")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), size=1, color="black") +
    geom_point(size=4, colour="black", fill="white") + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", size=0.5) + 
    xlab("") + 
    ylab("Gene expression (vst)")+
    geom_text(x=2, y=6.7, label="p=0.682", color="darkgray", size=4)+
    ylim(5.9,6.7)+
    theme_classic() + 
    geom_text(x=1.5, y=5.9, label="N=44 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      axis.text.x=element_text(angle=45, hjust=1, color="black"),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12), 
      strip.text=element_text(size=12, face="bold")
      ); glucose_transport_plot

ggsave(glucose_transport_plot, file="Mcap2020/Figures/TagSeq/GenomeV3/SpecificGenes/glucose_transport.png", width=6, height=6)
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Glucose Transport")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

## Lipid Metabolism 

Run a one way anova to test variation in expression across life stages. 
```{r}
specific_genes%>%
  filter(Function=="Lipid Metabolism")%>%
  
  aov(vst~lifestage, data=.)%>%
  summary()
```

```{r}
lipid_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Lipid Metabolism")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), size=1, color="black") +
    geom_point(size=4, colour="black", fill="white") + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", size=0.5) + 
    xlab("") + 
    ylab("Gene expression (vst)")+
    ylim(5.9,6.7)+
    theme_classic() + 
    geom_text(x=2, y=6.7, label="p<0.001", color="black", size=4)+
    geom_text(x=1.5, y=5.9, label="N=519 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      axis.text.x=element_text(angle=45, hjust=1, color="black"),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12), 
      strip.text=element_text(size=12, face="bold")
      ); lipid_plot

ggsave(lipid_plot, file="Mcap2020/Figures/TagSeq/GenomeV3/SpecificGenes/lipid_metabolism.png", width=6, height=6)
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Lipid Metabolism")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

## Fatty Acid Metabolism  

Run a one way anova to test variation in expression across life stages. 
```{r}
specific_genes%>%
  filter(Function=="Fatty Acid Metabolism")%>%
  
  aov(vst~lifestage, data=.)%>%
  summary()
```

```{r}
fattyacid_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Fatty Acid Metabolism")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), size=1, color="black") +
    geom_point(size=4, colour="black", fill="white") + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", size=0.5) + 
    xlab("") + 
    ylab("Gene expression (vst)")+
    geom_text(x=2, y=6.7, label="p<0.001", color="black", size=4)+
    ylim(5.9,6.7)+
    theme_classic() + 
    geom_text(x=1.5, y=5.9, label="N=159 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      axis.text.x=element_text(angle=45, hjust=1, color="black"),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12), 
      strip.text=element_text(size=12, face="bold")
      ); fattyacid_plot

ggsave(fattyacid_plot, file="Mcap2020/Figures/TagSeq/GenomeV3/SpecificGenes/fattyacid_metabolism.png", width=6, height=6)
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Fatty Acid Metabolism")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

## Ammonium Metabolism  

Run a one way anova to test variation in expression across life stages. 
```{r}
specific_genes%>%
  filter(Function=="Ammonium Metabolism")%>%
  
  aov(vst~lifestage, data=.)%>%
  summary()
```

```{r}
ammonium_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Ammonium Metabolism")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), size=1, color="black") +
    geom_point(size=4, colour="black", fill="white") + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", size=0.5) + 
    xlab("") + 
    ylab("Gene expression (vst)")+
    geom_text(x=2, y=6.7, label="p<0.001", color="black", size=4)+
    ylim(5.9,6.7)+
    theme_classic() + 
    geom_text(x=1.5, y=5.9, label="N=142 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      axis.text.x=element_text(angle=45, hjust=1, color="black"),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12), 
      strip.text=element_text(size=12, face="bold")
      ); ammonium_plot

ggsave(ammonium_plot, file="Mcap2020/Figures/TagSeq/GenomeV3/SpecificGenes/ammonium_metabolism.png", width=6, height=6)
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Ammonium Metabolism")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

## Oxidative Stress 

Run a one way anova to test variation in expression across life stages. 
```{r}
specific_genes%>%
  filter(Function=="Oxidative Stress")%>%
  
  aov(vst~lifestage, data=.)%>%
  summary()
```

```{r}
oxidative_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Oxidative Stress")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), size=1, color="black") +
    geom_point(size=4, colour="black", fill="white") + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", size=0.5) + 
    xlab("") + 
    ylab("Gene expression (vst)")+
  geom_text(x=2, y=6.7, label="p<0.001", color="black", size=4)+
    ylim(5.9,6.7)+
    theme_classic() + 
    geom_text(x=1.5, y=5.9, label="N=295 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      axis.text.x=element_text(angle=45, hjust=1, color="black"),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12), 
      strip.text=element_text(size=12, face="bold")
      ); oxidative_plot

ggsave(oxidative_plot, file="Mcap2020/Figures/TagSeq/GenomeV3/SpecificGenes/oxidative_stress.png", width=6, height=6)
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Oxidative Stress")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

## Bicarbonate Transport 

Run a one way anova to test variation in expression across life stages. 
```{r}
specific_genes%>%
  filter(Function=="Bicarbonate Transport")%>%
  
  aov(vst~lifestage, data=.)%>%
  summary()
```

```{r}
bicarb_plot<-specific_genes%>%
    group_by(Function, lifestage)%>%
    summarise(mean=mean(vst), N=length(vst), sd=sd(vst), sem=sd/N)%>%
    filter(Function=="Bicarbonate Transport")%>%
    ggplot(., aes(x = lifestage, y = mean)) +
    facet_wrap(~Function)+
    geom_line(aes(group=1), size=1, color="black") +
    geom_point(size=4, colour="black", fill="white") + 
  geom_errorbar(aes(ymin=mean-sem, ymax=mean+sem), width=0.1, linetype="solid", color="black", size=0.5) + 
    xlab("") + 
    ylab("Gene expression (vst)")+
  geom_text(x=2, y=7, label="p<0.001", color="black", size=4)+
    ylim(5.7,7)+
    theme_classic() + 
    geom_text(x=1.5, y=5.7, label="N=295 genes", size=4, fontface="italic")+
    theme(
      legend.position="none",
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      axis.text.x=element_text(angle=45, hjust=1, color="black"),
      legend.title=element_text(face="bold", size=14), 
      legend.text=element_text(size=12), 
      strip.text=element_text(size=12, face="bold")
      ); bicarb_plot

ggsave(bicarb_plot, file="Mcap2020/Figures/TagSeq/GenomeV3/SpecificGenes/bicarb_transport.png", width=6, height=6)
```

Display number of genes in this category.  
```{r}
specific_genes%>%
  filter(Function=="Bicarbonate Transport")%>%
  group_by(Function)%>%
  summarise(Unique_Elements = n_distinct(gene_id))
```

# Assemble all plots together  
```{r}
library(cowplot)

#no x axis label
carb_plot #with y axis 
glucose_plot
glucose_transport_plot
lipid_plot

#with x axis labels 
bicarb_plot # with y axis 
fattyacid_plot
ammonium_plot
oxidative_plot

carb_plot2<-carb_plot+theme(axis.text.x=element_blank())
glucose_plot2<-glucose_plot+theme(axis.text.x=element_blank(), axis.title.y=element_blank())
glucose_transport_plot2<-glucose_transport_plot+theme(axis.text.x=element_blank(), axis.title.y=element_blank())
lipid_plot2<-lipid_plot+theme(axis.text.x=element_blank(), axis.title.y=element_blank())
bicarb_plot2<-bicarb_plot
fattyacid_plot2<-fattyacid_plot+theme(axis.title.y=element_blank())
ammonium_plot2<-ammonium_plot+theme(axis.title.y=element_blank())
oxidative_plot2<-oxidative_plot+theme(axis.title.y=element_blank())

gene_plots<-plot_grid(carb_plot2, glucose_plot2, glucose_transport_plot2, lipid_plot2, bicarb_plot2, fattyacid_plot2, ammonium_plot2, oxidative_plot2, ncol=4, nrow=2, rel_heights = c(0.7,1))
                      
ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/SpecificGenes/gene_expression_class.jpg", plot=gene_plots, dpi=300, width=22, height=9, units="in")
ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/SpecificGenes/gene_expression_class.pdf", plot=gene_plots, dpi=300, width=22, height=9, units="in")
```







