---
title: "DESeq2 analysis of M. capitata TagSeq"
author: "Ariana S Huffmyer"
date: "2024"
output: html_document
editor_options: 
  chunk_output_type: console
---

This script uses the V3 genome from Cyanophora Rutgers.  

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# The following setting is important, do not omit.
options(stringsAsFactors = FALSE) #Set Strings to character
```

# Load required libraries    

```{r}
if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse') 
if ("genefilter" %in% rownames(installed.packages()) == 'FALSE') install.packages('genefilter') 
if ("DESeq2" %in% rownames(installed.packages()) == 'FALSE') install.packages('DESeq2') 
if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer') 
if ("WGCNA" %in% rownames(installed.packages()) == 'FALSE') install.packages('WGCNA') 
if ("flashClust" %in% rownames(installed.packages()) == 'FALSE') install.packages('flashClust') 
if ("gridExtra" %in% rownames(installed.packages()) == 'FALSE') install.packages('gridExtra') 
if ("ComplexHeatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('ComplexHeatmap') 
if ("goseq" %in% rownames(installed.packages()) == 'FALSE') install.packages('goseq') 
if ("dplyr" %in% rownames(installed.packages()) == 'FALSE') install.packages('dplyr') 
if ("clusterProfiler" %in% rownames(installed.packages()) == 'FALSE') install.packages('clusterProfiler') 
if ("pheatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('pheatmap') 
if ("magrittr" %in% rownames(installed.packages()) == 'FALSE') install.packages('magrittr') 
if ("rtracklayer" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rtracklayer")
if ("GenomicRanges" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GenomicRanges")
if ("plyranges" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("plyranges")
if ("GSEABase" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GSEABase")
if ("GOSim" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GOSim")
if ("stats" %in% rownames(installed.packages()) == 'FALSE') install.packages("stats")
if ("ggdendro" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("ggdendro")
#if ("KEGG.db" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("KEGG.db")
if ("GO.db" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GO.db")
if ("rrvgo" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rrvgo")

#BiocManager::install("org.Ce.eg.db", force=TRUE) #install if needed 

library("tidyverse")
library("genefilter")
library("DESeq2")
library("RColorBrewer")
library("WGCNA")
library("flashClust")
library("gridExtra")
library("ComplexHeatmap")
library("goseq")
library("dplyr")
library("clusterProfiler")
library("pheatmap")
library("magrittr")
library("rtracklayer")
library("GenomicRanges")
library("plyranges")
library("GSEABase")
library("GOSim")
library("stats")
library("ggdendro")
#library("KEGG.db")
library("GO.db")
library("rrvgo")
library("DESeq2")
library("genefilter")
library("factoextra")
library("vegan")
library("EnhancedVolcano")
library("VennDetail")
library(viridis) 
library(cowplot)
```

# Data input and filtering 
 
Import the data files.  

```{r}
#load metadata sheet with sample name and developmental stage information
metadata <- read.csv("Mcap2020/Data/TagSeq/Sample_Info.csv", header = TRUE, sep = ",")
head(metadata)

#load gene count matrix generated from cluster computation
gcount <- as.data.frame(read.csv("Mcap2020/Output/TagSeq/Mcapitata_gene_count_matrix_V3.csv", row.names="gene_id"), colClasses = double)
head(gcount)

#remove metamorphosed recruit 1 because this was the timepoint that we only had one sample for: D36, AH23
#gcount<-gcount[ , !(names(gcount) %in% c("AH23"))]
gcount<-gcount[ , !(names(gcount) %in% c("AH23_S54_L002.gtf"))]

metadata <- metadata%>%
  filter(!sample_id=="AH23")
```

Check that there are no genes with 0 counts across all samples.

```{r}
nrow(gcount)
gcount<-gcount %>%
     mutate(Total = rowSums(.[, 1:38]))%>%
    filter(!Total==0)%>%
    dplyr::select(!Total)
nrow(gcount)
```
We had 54,384 genes, which was filtered down to 34,223 by removing genes with row sums of 0 (those not detected in our data).  
 
Conduct data filtering, this includes:  

*pOverA*: Specifying the minimum count for a proportion of samples for each gene. Here, we are using a pOverA of 0.07. This is because we have 38 samples with a minimum of n=3 samples per lifestage. Therefore, we will accept genes that are present in 3/38 = 0.07 of the samples because we expect different expression by life stage. We are further setting the minimum count of genes to 10, such that 7% of the samples must have a gene count of >10 in order for the gene to remain in the data set.  

Filter in the package "genefilter". Pre-filtering our dataset to reduce the memory size dataframe, increase the speed of the transformation and testing functions, and improve quality of statistical analysis by removing low-coverage counts. Removed counts could represent outliers in the data and removing these improves sensitivity of statistical tests.   

```{r}
filt <- filterfun(pOverA(0.07,10))

#create filter for the counts data
gfilt <- genefilter(gcount, filt)

#identify genes to keep by count filter
gkeep <- gcount[gfilt,]

#identify genes to keep by count filter
gkeep <- gcount[gfilt,]

#identify gene lists
gn.keep <- rownames(gkeep)

#gene count data filtered in PoverA, P percent of the samples have counts over A
gcount_filt <- as.data.frame(gcount[which(rownames(gcount) %in% gn.keep),])

#How many rows do we have before and after filtering?
nrow(gcount) #Before
nrow(gcount_filt) #After
```

Before filtering, we had 34,223 genes. After filtering for pOverA, we have approximately 11,475 genes. This indicates that there were many genes present in <7% of samples at <10 counts per gene.  

In order for the DESeq2 algorithms to work, the SampleIDs on the metadta file and count matrices have to match exactly and in the same order. The following R clump will check to make sure that these match. Should return TRUE.  
```{r}
#Checking that all row and column names match. Should return "TRUE"
all(rownames(metadata$sample_id) %in% colnames(gcount_filt))
all(rownames(metadata$sample_id) == colnames(gcount_filt)) 
```

Match metadata sample id's.  
```{r}
result<-sub("_.*", "", colnames(gcount_filt)) #keep characters before _
#result2<-sub(".*AH", "", result) #keep characters after X
colnames(gcount_filt)<-result #keep just sample number 
```

Display current order of metadata and gene count matrix.  
```{r}
metadata$sample_id
colnames(gcount_filt)
```

Order metadata the same as the column order in the gene matrix.  
```{r}
list<-colnames(gcount_filt)
list<-as.factor(list)

metadata$sample_id<-as.factor(metadata$sample_id)

# Re-order the levels
metadata$sample_id <- factor(as.character(metadata$sample_id), levels=list)
# Re-order the data.frame
metadata_ordered <- metadata[order(metadata$sample_id),]
metadata_ordered$sample_id

metadata_ordered<-metadata_ordered%>%
  dplyr::select(sample_id, lifestage, hpf, code)
```

Reorder the lifestage factor.  
```{r}
head(metadata_ordered)
levels(as.factor(metadata_ordered$lifestage))

metadata_ordered$lifestage<-as.factor(metadata_ordered$lifestage)

metadata_ordered$lifestage<-fct_relevel(metadata_ordered$lifestage, "Egg (1 hpf)", "Embryo (5 hpf)", "Embryo (38 hpf)", "Embryo (65 hpf)", "Larvae (93 hpf)", "Larvae (163 hpf)", "Larvae (231 hpf)", "Metamorphosed Polyp (231 hpf)", "Attached Recruit (183 hpf)", "Attached Recruit (231 hpf)")

levels(as.factor(metadata_ordered$lifestage))

#output metadata for other analyses 
#write_csv(metadata_ordered, "Mcap2020/Data/TagSeq/metadata_ordered.csv")
```

# Construct DESeq2 data set  

Create a DESeqDataSet design from gene count matrix and labels. Here we set the design to look at lifestage to test for any differences in gene expression across timepoints.
```{r}
#Set DESeq2 design
gdds <- DESeqDataSetFromMatrix(countData = gcount_filt,
                              colData = metadata_ordered,
                              design = ~lifestage)
```

First we are going to log-transform the data using a variance stabilizing transforamtion (VST). This is only for visualization purposes. Essentially, this is roughly similar to putting the data on the log2 scale. It will deal with the sampling variability of low counts by calculating within-group variability (if blind=FALSE). Importantly, it does not use the design to remove variation in the data, and so can be used to examine if there may be any variability do to technical factors such as extraction batch effects.

To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4.

Chunk should return TRUE if <4.  
```{r}
SF.gdds <- estimateSizeFactors(gdds) #estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than 4 for us to use vst
print(sizeFactors(SF.gdds)) #View size factors

all(sizeFactors(SF.gdds)) < 4
```

All size factors are less than 4, so we can use VST transformation.  

```{r}
gvst <- vst(gdds, blind=FALSE) #apply a variance stabilizing transformation to minimize effects of small counts and normalize wrt library size
head(assay(gvst), 3) #view transformed gene count data for the first three genes in the dataset.  
```

# Conduct PERMANOVA for all genes  

Export data for PERMANOVA test.  
```{r}
test<-t(assay(gvst)) #export as matrix
test<-as.data.frame(test)

#add category columns
test$Sample<-rownames(test)
test$Lifestage<-metadata$lifestage[match(test$Sample, metadata$sample_id)]
```

Build PERMANOVA model.  
```{r}
scaled_test <-prcomp(test[c(1:11475)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test)

# scale data
vegan <- scale(test[c(1:11475)])

# PerMANOVA 
permanova<-adonis2(vegan ~ Lifestage, data = test, method='eu')
permanova
```
40% variance explained on PC1, with 10% or less by remaining PCs.  

Gene expression is significantly different between lifestages (p=0.001).  

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegan ~ Lifestage, data = test, method = "eu")
          Df SumOfSqs      R2      F Pr(>F)    
Lifestage  9   318674 0.75057 9.3618  0.001 ***
Residual  28   105901 0.24943                  
Total     37   424575 1.00000     

# PCA and sample distances of all genes  

Plot a heatmap to sample to sample distances  

```{r}
gsampleDists <- dist(t(assay(gvst))) #calculate distance matix
gsampleDistMatrix <- as.matrix(gsampleDists) #distance matrix
rownames(gsampleDistMatrix) <- colnames(gvst) #assign row names
colnames(gsampleDistMatrix) <- NULL #assign col names
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255) #assign colors

save_pheatmap_pdf <- function(x, filename, width=7, height=7) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}

pht<-pheatmap(gsampleDistMatrix, #plot matrix
         clustering_distance_rows=gsampleDists, #cluster rows
         clustering_distance_cols=gsampleDists, #cluster columns
         col=colors) #set colors

#save_pheatmap_pdf(pht, "Mcap2020/Figures/TagSeq/GenomeV3/pheatmap.pdf")
```

Generate a PCA for visualization/presentations.  

```{r}
gPCAdata <- plotPCA(gvst, intgroup = c("lifestage"), returnData=TRUE, ntop=11475) #use ntop to specify all genes
percentVar <- round(100*attr(gPCAdata, "percentVar")) #plot PCA of samples with all data

levels(gPCAdata$lifestage)

allgenesfilt_PCA_visual <- 
  ggplot(data=gPCAdata, aes(PC1, PC2, shape=lifestage, colour=lifestage)) + 
  geom_point(size=5) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  
  scale_shape_manual(name="Lifestage", values = c("Egg (1 hpf)"=17, "Embryo (5 hpf)"=16, "Embryo (38 hpf)"=15, "Embryo (65 hpf)"=17, "Larvae (93 hpf)"=15, "Larvae (163 hpf)"=16, "Larvae (231 hpf)"=17, "Metamorphosed Polyp (231 hpf)"=16, "Attached Recruit (183 hpf)"=15, "Attached Recruit (231 hpf)"=16)) +
  
  scale_colour_manual(name="Lifestage", values = c("Egg (1 hpf)"="#8C510A", "Embryo (5 hpf)"="#DFC27D", "Embryo (38 hpf)"="#DFC27D", "Embryo (65 hpf)"="#DFC27D", "Larvae (93 hpf)"="#80CDC1", "Larvae (163 hpf)"="#80CDC1", "Larvae (231 hpf)"="#80CDC1", "Metamorphosed Polyp (231 hpf)"="#003C30", "Attached Recruit (183 hpf)"="#BA55D3", "Attached Recruit (231 hpf)"="#BA55D3")) +
  #xlim(-40,40)+ 
  #ylim(-40,40)+
  coord_fixed()+
  theme_classic() + #Set background color
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) # + #Set the plot background
  #theme(legend.position = ("none")) #set title attributes
allgenesfilt_PCA_visual

#ggsave("Mcap2020/Figures/TagSeq/GenomeV3/allgenesfilt-PCA_visual.pdf", allgenesfilt_PCA_visual, width=11, height=8)
```

# Run DESeq Results Tests 

Run test using the LRT to look for effect of lifestage as a whole across our experiment. There is only one main effect, so the reduced model is just the intercept.    
```{r, message = FALSE}
DEG <- DESeq(gdds, test="LRT", reduced=~1)
```

From this, we are going to filter by adjusted p value <0.05, this is generated by the LRT test statistic. We had 9180 DEGs at P<0.05. We are also running at P<0.01 because we need a higher threshold due to the high number of DEGs. This was only reduced to 8686 with P<0.01. It was reduced further to P<0.001 with 8127 genes. We will keep this for now.  

```{r}
DEG_results<-results(DEG)
DEG_results <- as.data.frame(subset(DEG_results, padj<0.001))

DEG_results$gene <- rownames(DEG_results)
rownames(DEG_results) <- NULL
```
There are 8127 DEGs out of 11,475 total genes (70.8%). Most genes are differentially expressed across time. This is expected for a developmental time series spanning eggs to recruits due to the drastic changes in function.  

Conduct VST transformation.  
```{r}
DEG_results_vst <- gdds[unique(DEG_results$gene)]
dim(DEG_results_vst)

DEG_results_vst <- varianceStabilizingTransformation(DEG_results_vst) 
```

# Plot heatmap of DEGs 

Set themes and metadata.  
```{r}
legend_names_col = colnames(assay(DEG_results_vst))

df <- as.data.frame(colData(gdds)[,c("lifestage")])
names(df)<-"lifestage"
rownames(df) <- legend_names_col

levels(df$lifestage)

df$lifestage <- factor(as.character(df$lifestage))

ann_colors = list( 
    lifestage = c("Egg (1 hpf)"="#8C510A", "Embryo (5 hpf)"="tan2", "Embryo (38 hpf)"="tan3",  "Embryo (65 hpf)"="tan", "Larvae (93 hpf)"="#80CDC1", "Larvae (163 hpf)"="cyan2", "Larvae (231 hpf)"="cyan", "Metamorphosed Polyp (231 hpf)"="#003C30", "Attached Recruit (183 hpf)"="#BA55D3", "Attached Recruit (231 hpf)"="pink3")
)
```

Plot heatmap.  
```{r}
pdf("Mcap2020/Figures/TagSeq/GenomeV3/DESeq/DEG_heatmap_development.pdf", width=15, height=20)
pheatmap(assay(DEG_results_vst), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
dev.off()
```

Eggs through 38 hpf group together; 65-163 hpf group together, attached recruits group together, and larvae and polyps at 231 hpf group together. 

# PERMANOVA and PCA of DEGs 

Conduct PERMANOVA by parent and temperature 

Export data for PERMANOVA test.  
```{r}
test_DEG<-t(assay(DEG_results_vst)) #export as matrix
test_DEG<-as.data.frame(test_DEG)

#add category columns
test_DEG$sample<-rownames(test_DEG)
test_DEG$lifestage<-metadata$lifestage[match(test_DEG$sample, metadata$sample)]
```

Build PERMANOVA model for DEG's that are different between temperatures.  
```{r}
dim(test_DEG)

scaled_test_DEG <-prcomp(test_DEG[c(1:8127)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test_DEG)

# scale data
vegan_DEG <- scale(test_DEG[c(1:8127)])

# PerMANOVA 
permanova_DEG<-adonis2(vegan_DEG ~ lifestage, data = test_DEG, method='eu')
permanova_DEG
```
About 50% of variance is explained by first PC followed by 10% or less on following PCs. 

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegan_DEG ~ lifestage, data = test_DEG, method = "eu")
          Df SumOfSqs     R2      F Pr(>F)    
lifestage  9   250210 0.8321 15.418  0.001 ***
Residual  28    50489 0.1679                  
Total     37   300699 1.0000   

Plot PCA.  
```{r}
gPCAdata_DEG <- plotPCA(DEG_results_vst, intgroup = c("lifestage"), returnData=TRUE, ntop=8127) #use ntop to specify all genes
percentVar_DEG <- round(100*attr(gPCAdata_DEG, "percentVar")) #plot PCA of samples with all data

levels(gPCAdata$lifestage)

DEG_PCA <- 
  ggplot(data=gPCAdata_DEG, aes(PC1, PC2, shape=lifestage, colour=lifestage)) + 
  geom_point(size=5, stroke=1) +
  xlab(paste0("PC1: ",percentVar_DEG[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_DEG[2],"% variance")) +
  
  scale_shape_manual(name="Lifestage", values = c("Egg (1 hpf)"=1, "Embryo (5 hpf)"=2, "Embryo (38 hpf)"=3, "Embryo (65 hpf)"=4, "Larvae (93 hpf)"=5, "Larvae (163 hpf)"=6, "Larvae (231 hpf)"=8, "Metamorphosed Polyp (231 hpf)"=11, "Attached Recruit (183 hpf)"=10, "Attached Recruit (231 hpf)"=12)) +
  
  scale_colour_manual(name="Lifestage", values = c("Egg (1 hpf)"="#8C510A", "Embryo (5 hpf)"="#DFC27D", "Embryo (38 hpf)"="#DFC27D", "Embryo (65 hpf)"="#DFC27D", "Larvae (93 hpf)"="#80CDC1", "Larvae (163 hpf)"="#80CDC1", "Larvae (231 hpf)"="#80CDC1", "Metamorphosed Polyp (231 hpf)"="#003C30", "Attached Recruit (183 hpf)"="#BA55D3", "Attached Recruit (231 hpf)"="#BA55D3")) +

  #coord_fixed()+
  theme_classic() + #Set background color
  theme(panel.border = element_rect(colour = "black", fill=NA), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank(), 
        axis.title=element_text(size=14), 
        axis.text=element_text(size=12, color="black"))  + #Set the plot background
  theme(legend.position = ("right")); DEG_PCA

ggsave("Mcap2020/Figures/TagSeq/GenomeV3/DESeq/PCA_DEG.pdf", DEG_PCA, width=8, height=5)
```

# Save gene lists for functional enrichment 

## lifestage genes

DEG_results: List of DEG's and fold change/pvalues
DEG_results_vst: Count matrix of genes from DEG list 

8127 genes 

```{r}
dim(DEG_results)
head(DEG_results)

deg_results_list<-DEG_results%>%
  dplyr::select(gene, everything())
write_csv(deg_results_list, "Mcap2020/Output/TagSeq/GenomeV3/DESeq/deg_list.csv")

dim(DEG_results_vst)
deg_counts<-t(assay(DEG_results_vst))
deg_counts<-as.data.frame(deg_counts)
dim(deg_counts)
deg_counts$sample<-rownames(deg_counts)
deg_counts<-deg_counts%>%
  dplyr::select(sample, everything())
write_csv(deg_counts, "Mcap2020/Output/TagSeq/GenomeV3/DESeq/deg_counts.csv")
```

## All genes detected 

List of all genes in the dataset (gvst) = 11475 genes which will serve as a reference for functional enrichment.   

```{r}
dim(gvst)
all_genes<-t(assay(gvst))
all_genes<-as.data.frame(all_genes)
dim(all_genes)
all_genes$sample<-rownames(all_genes)
all_genes<-all_genes%>%
  dplyr::select(sample, everything())

write_csv(file="Mcap2020/Output/TagSeq/GenomeV3/DESeq/all_genes.csv", all_genes)
```

# Detect clusters of LRT results   

code from https://hbctraining.github.io/DGE_workshop_salmon_online/lessons/08a_DGE_LRT_results.html 
```{r}
# results 
DEG <- DESeq(gdds, test="LRT", reduced=~1)

# Extract results for LRT
res_LRT <- results(DEG)

# View results for LRT
res_LRT  
```

Filter for genes padj <0.001. For LRT, the p-value is the cut off associated with the test statistic. Fold change is not associated with the statistical test. It can be viewed, but you cannot select genes based on a fold change cut off to determine DEGs.  

```{r}
padj.cutoff <- 0.001

# Create a tibble for LRT results
res_LRT_tb <- res_LRT %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()

# Subset to return genes with padj < target
sigLRT_genes <- res_LRT_tb %>% 
  dplyr::filter(padj < padj.cutoff)

# Get number of significant genes
nrow(sigLRT_genes)
```
There are 8127 genes that change across developmental time at p<0.001.    

Perform clustering. 

```{r}
library(DEGreport)

clustering_sig_genes <- sigLRT_genes %>%
  arrange(padj) 

matrix<-assay(gvst)

meta <- metadata_ordered %>%
  dplyr::select(sample_id, lifestage)

rownames(meta)<-meta$sample_id

meta<-meta%>%
  dplyr::select(lifestage)

# Obtain transformed values for those significant genes
cluster_vst <- matrix[clustering_sig_genes$gene, ]
```

Run with minimum 100 genes per cluster to detect large clusters because we have many DEGs. This will take awhile to run.  
```{r}
# Use the `degPatterns` function from the 'DEGreport' package to show gene clusters across sample groups
clusters <- degPatterns(cluster_vst, metadata = meta, time = "lifestage", col=NULL, plot=FALSE, minc = 100)

```
6372 genes were clustered with cluster size >100 

Plot is showing Z score of VST normalized gene counts  
```{r}
dev.off()
cluster_plot<-ggplot(clusters[["normalized"]],
        aes(x=lifestage, y=value)) +
  facet_wrap(~ cluster)+
   geom_point(colour="darkgray", size=1) +
  ylab("Z-score of VST normalized gene counts")+
  xlab("")+
   # change the method to make it smoother
   geom_smooth(aes(group=genes), method = "loess", colour="red", linewidth=0.1, se=FALSE)+
   geom_smooth(aes(group=1), method = "loess", colour="black")+
  geom_hline(yintercept=0, linetype="dashed")+
  geom_vline(xintercept=3, linetype="dotted")+
  geom_vline(xintercept=6, linetype="dotted")+
  theme_classic()+
  theme(
    axis.text.x = element_text(angle=90, hjust=1, vjust=1)
  );cluster_plot

ggsave(cluster_plot, filename="Mcap2020/Figures/TagSeq/GenomeV3/DESeq/cluster_plot.pdf", width=10, height=8)
```

```{r}
# Rename and extract information 
group <- clusters$df %>%
          rename(gene=genes)
``` 

We can perform functional enrichment on these clusters now.  

Assign developmental pattern. DEG PCA and heatmaps showed three main clusters: Early development (eggs through 38 hpf), mid development (65-163 hpf), and late development (183-231 hpf). Assign cluster to each of these three patterns by looking at where the peak occurs in the above panel of plots.  

```{r}
group <- group%>%
  mutate(pattern=if_else(cluster=="1", "Late", 
                         if_else(cluster=="2", "Late", 
                                 if_else(cluster=="3", "Mid", 
                                         if_else(cluster=="5", "Late", 
                                                 if_else(cluster=="6", "Late", 
                                                         if_else(cluster=="7", "Early", 
                                                                 if_else(cluster=="11", "Early", 
                                                                         if_else(cluster=="12", "Early", 
                                                                                 if_else(cluster=="14", "Mid", 
                                                                                         if_else(cluster=="15", "Mid", 
                                                                                                 if_else(cluster=="16", "Mid", 
                                                                                                         if_else(cluster=="17", "Late", 
                                                                                                                 if_else(cluster=="18", "Early", 
                                                                                                                         if_else(cluster=="19", "Early", 
                                                                                                                                 if_else(cluster=="20", "Mid", 
                                                                                                                                         if_else(cluster=="21", "Early", 
                                                                                                                                                 if_else(cluster=="24", "Early", NA))))))))))))))))))
```

Display number of genes in each cluster. 
```{r}
ngenes<-clusters$summarise%>%
  as.data.frame()

ngenes<-ngenes%>%dplyr::select(cluster, n_genes)%>%unique()

ngenes

#early development 
#7, 11, 12, 18, 19, 21, 24
211+464+443+269+229+117+154
#1887 genes in early development clusters 

#mid development 
#3, 14, 15, 16, 20
504+444+104+404+240
#1696 genes in mid development 

#late development
#1, 2, 5, 6, 17
608+826+390+606+359
#2789 genes in late development 
```

Plot fold change. 

```{r}
deg_FC <- DEG_results %>% 
  dplyr::arrange(log2FoldChange) %>%
  
  ggplot(aes(x=log2FoldChange, y=reorder(gene, -log2FoldChange), colour=padj)) +
  #facet_grid(~contrast, scales="free_y")+
  geom_point(size=1)+
  scale_colour_gradientn(name="P(adj)", colours = colorspace::heat_hcl(7))+
  geom_segment(aes(x=0, xend=log2FoldChange, y=gene, yend=gene),linewidth=1) +
  geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("Gene")+
  theme(legend.position = "right",
        axis.text.y = element_blank(),
        axis.text.x = element_text(size=14),
        strip.text=element_text(size=16),
        axis.title.y = element_text(size=16, face="bold"),
        axis.title.x = element_text(size = 16, face = "bold"),
        plot.title = element_text(size = 18, face = "bold"))+
  ggtitle(''); deg_FC

ggsave(plot=deg_FC, "Mcap2020/Figures/TagSeq/GenomeV3/DESeq/DEG_foldchange.png", height=6, width=8)
```

# Functional enrichment of genes identified in clusters

## Prepare functional enrichment dataframes 

Read in files 

```{r}
all_genes<-read.csv("Mcap2020/Output/TagSeq/GenomeV3/DESeq/all_genes.csv")
```

Read in file of DEG's 
```{r}
DEG_list<-read.csv("Mcap2020/Output/TagSeq/GenomeV3/DESeq/deg_list.csv")
DEG_counts<-read.csv("Mcap2020/Output/TagSeq/GenomeV3/DESeq/deg_counts.csv")
```

Get gene length information.  
```{r}
#import file
gff <- rtracklayer::import("Mcap2020/Data/TagSeq/Montipora_capitata_HIv3.genes.gff3") #if this doesn't work, restart R and try again 

transcripts <- subset(gff, type == "transcript") #keep only transcripts 

transcripts_gr <- makeGRangesFromDataFrame(transcripts, keep.extra.columns=TRUE) #extract length information

transcript_lengths <- width(transcripts_gr) #isolate length of each gene

seqnames<-transcripts_gr$ID #extract list of gene id 

lengths<-cbind(seqnames, transcript_lengths)

lengths<-as.data.frame(lengths) #convert to data frame
```

Read in annotation file. 

Downloaded functional annotation (version 3) from: http://cyanophora.rutgers.edu/montipora/Montipora_capitata_HIv3.genes.EggNog_results.txt.gz (EggNog) and http://cyanophora.rutgers.edu/montipora/Montipora_capitata_HIv3.genes.KEGG_results.txt.gz (KEGG). 

Unzipped on desktop. I then removed the # in #query in the first column, otherwise it does not read in column names.  

```{r}
Mcap.annot <- read.table("Mcap2020/Data/TagSeq/Montipora_capitata_HIv3.genes.EggNog_results.txt",  quote="", sep="\t", header=TRUE)
dim(Mcap.annot)

# column names
head(Mcap.annot)
tail(Mcap.annot)

Mcap.annot<-Mcap.annot%>%
  rename(gene=query)

names(Mcap.annot)
```

This functional annotation has 24,072 genes. This is 44% of total protein predicted genes described in the publication associated with this annotation. https://academic.oup.com/gigascience/article/doi/10.1093/gigascience/giac098/6815755 

Match up genes in gene list file to annotation file
```{r}
names(Mcap.annot)

#remove sample column name 
probes = names(all_genes)
#probes <- gsub("sample", "", probes)
#probes <- probes[probes != ""]

probes2annot = match(probes, Mcap.annot$gene) #match genes in datExpr to genes in annotation file, note I removed the # before query in this file before loading

# The following is the number of probes without annotation 
sum(is.na(probes2annot))

row_nas<-which(is.na(probes2annot))

#view the genes that do not have a match in the annotation file
missing<-as.data.frame(probes[row_nas])
#print(missing)
```

There are 3395 of our genes that were expressed that are not present in the annotation file.  

Reduce annotation file to only contain genes detected in our dataset.  

```{r}
filtered_Mcap.annot <- Mcap.annot[Mcap.annot$gene %in% probes, ]
dim(filtered_Mcap.annot)
```

The annotation file now only contains genes that were detected in our dataset that have annotation information. This is 8081 out of the 11475 genes in our dataset. 

Add in length to the annotation file.    

```{r}
filtered_Mcap.annot$length<-lengths$transcript_lengths[match(filtered_Mcap.annot$gene, lengths$seqnames)]

which(is.na(filtered_Mcap.annot$length)) #all genes have lengths 
```

All have length information.  

Format GO terms to remove dashes and quotes.  
```{r}
#remove dashes, remove quotes from columns and separate by semicolons (replace , with ;) in KEGG_ko, KEGG_pathway, and Annotation.GO.ID columns
head(filtered_Mcap.annot$GOs)
filtered_Mcap.annot$GOs <- gsub(",", ";", filtered_Mcap.annot$GOs)
filtered_Mcap.annot$GOs <- gsub('"', "", filtered_Mcap.annot$GOs)
filtered_Mcap.annot$GOs <- gsub("-", NA, filtered_Mcap.annot$GOs)
```

## Conduct enrichment for individual clusters 

Conduct enrichment for clusters 

### Conduct enrichment for each cluster  

Merge in relevant annotation information to the DEG list file.  

Merge to only keep genes that were identified to belong to a cluster. 
```{r}
group_list<-left_join(group, filtered_Mcap.annot, by="gene")
```

Conduct functional enrichment using goseq and rrvgo. Compare against all genes detected in the dataset. 







DOUBLE CHECK ID VECTOR FOR PWF FUNCTION - ALL GENES OR JUST DEGS??????

Should ID vector be ALL.vector instead in nullp and goseq???? 


MAYBE DONT EVEN NEED THE ALL VECTOR/ID VECTOR HERE?? 
https://github.com/fishercera/TreehopperSeq/blob/master/GoSeq_Walkthrough.md





Conduct at biological process level. Filter for adjusted p-value <0.05.  
```{r}
cluster_list<-levels(as.factor(group_list$cluster))

for (cluster_id in cluster_list) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(filtered_Mcap.annot$gene)
  
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(filtered_Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- group_list%>%
      filter(cluster==cluster_id)%>%
      pull(gene)
    
    ##Get a list of GO Terms for all genes detected in our dataset 
    GO.terms <- filtered_Mcap.annot%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene: load in the gene.vector(a list of all genes with 1 indicating it is a DEG in the group of interest and 0 meaning it is not a DEG) and bias.data (list of lengths for all genes)
    pwf<-nullp(gene.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms: load in the pwf object (gene name, a 1 or 0 to indicate DEG, the length, and the pwf function; generated from pwf above), and the list of GO terms for all genes
    GO.wall<-goseq(pwf, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <- p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.05
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("Mcap2020/Output/TagSeq/GenomeV3/DESeq/functional_enrichment/goseq_cluster", cluster_id, ".csv"))
    
}
```

```{r}
length(ALL.vector)
length(LENGTH.vector)
dim(pwf)
length(unique(GO.terms$gene))
```
All the dimensions of the input dataframes count to 8081, the expected number of genes. 

Clusters 6 (steady increase), 7 (higest in eggs/embryos), 16 (peak in larvae), 17 (peak in larvae with second peak in recruits), 19 (decrease), 20 (peak in larvae), 21 (decrease), 24 (peak in embryos) do not have any significant GO terms. 

Clusters 1, 2, 3, 5, 11, 12, 14, 15, 18 have significantly enriched GO terms (9 clusters). 

Create an empty dataframe to populate (only do this once).  

```{r}
cluster_go_results <- data.frame()
```

Run a loop to filter enrichment for each temperature contrast using rrvgo. 

clusters 2 and 3 dont have enough enriched go terms to get past the rrvgo step - these are not included. 
```{r}

cluster_list<-c("1", "5", "11", "12", "14", "15", "18")

for (cluster_id in cluster_list) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("Mcap2020/Output/TagSeq/GenomeV3/DESeq/functional_enrichment/goseq_cluster", cluster_id, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      #filter(numInCat>5)%>%
      arrange(., bh_adjust)
    
    print(cluster_id)
    print(length(unique(go_results$GOterm)))
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #calculate similarity 
    scores <- setNames(-log10(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$cluster<-cluster_id
    
    print(length(unique(go_results$GOterm)))
    
    #add to data frame 
    go_df <- go_results
    cluster_go_results <- rbind(cluster_go_results, go_df) 

}
```

Semantic similarity reduced GO terms for each cluster.    

```{r}
length(unique(cluster_go_results$GOterm))
```
There are 458 total enriched GO terms detected.  

### Plot enrichment for each cluster

Clusters 1, 5, 11, 12, 14, 15, 18 have significantly enriched GO terms after rrvgo (7 clusters). 

Cluster 1 (peaking in larvae through recruits)
```{r}
go_plot_cluster1 <- cluster_go_results %>% 
  filter(cluster=="1")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 25, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("GO Term")+
  xlab("Hits (%)")+
  ggtitle("Cluster 1 (larvae-recruit)")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"), 
        strip.text.y.right=element_text(angle=0)); go_plot_cluster1

ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/DESeq/functional_enrichment/go_cluster1.png", go_plot_cluster1, width=10, height=10)
```

Cluster 5 (peaking in larvae through polyp)
```{r}
go_plot_cluster5 <- cluster_go_results %>% 
  filter(cluster=="5")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 25, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("GO Term")+
  xlab("Hits (%)")+
  ggtitle("Cluster 5 (larvae-polyp)")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"), 
        strip.text.y.right=element_text(angle=0)); go_plot_cluster5

ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/DESeq/functional_enrichment/go_cluster5.png", go_plot_cluster5, width=10, height=18)
```

Cluster 11 (peaking in egg through embryo 38 hpf)
```{r}
go_plot_cluster11 <- cluster_go_results %>% 
  filter(cluster=="11")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 25, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("GO Term")+
  xlab("Hits (%)")+
  ggtitle("Cluster 11 (egg-embryo)")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"), 
        strip.text.y.right=element_text(angle=0)); go_plot_cluster11

ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/DESeq/functional_enrichment/go_cluster11.png", go_plot_cluster11, width=10, height=26)
```

Cluster 12 (peaking in egg through embryo 38 hpf)
```{r}
go_plot_cluster12 <- cluster_go_results %>% 
  filter(cluster=="12")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 25, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("GO Term")+
  xlab("Hits (%)")+
  ggtitle("Cluster 12 (egg-embryo)")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"), 
        strip.text.y.right=element_text(angle=0)); go_plot_cluster12

ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/DESeq/functional_enrichment/go_cluster12.png", go_plot_cluster12, width=10, height=8)
```

Cluster 14 (peaking in 65 hpf embryo to polyps)
```{r}
go_plot_cluster14 <- cluster_go_results %>% 
  filter(cluster=="14")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 25, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("GO Term")+
  xlab("Hits (%)")+
  ggtitle("Cluster 14 (embryo-polyp)")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"), 
        strip.text.y.right=element_text(angle=0)); go_plot_cluster14

ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/DESeq/functional_enrichment/go_cluster14.png", go_plot_cluster14, width=10, height=10)
```

Cluster 15 (peaking in 38 hpf embryo to 93 hpf larvae)
```{r}
go_plot_cluster15 <- cluster_go_results %>% 
  filter(cluster=="15")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 25, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("GO Term")+
  xlab("Hits (%)")+
  ggtitle("Cluster 15 (embryo-larvae)")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"), 
        strip.text.y.right=element_text(angle=0)); go_plot_cluster15

ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/DESeq/functional_enrichment/go_cluster15.png", go_plot_cluster15, width=10, height=8)
```

Cluster 18 (peaking in egg through 65 hpf embryo)
```{r}
go_plot_cluster18 <- cluster_go_results %>% 
  filter(cluster=="18")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 25, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("GO Term")+
  xlab("Hits (%)")+
  ggtitle("Cluster 18 (egg-embryo)")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"), 
        strip.text.y.right=element_text(angle=0)); go_plot_cluster18

ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/DESeq/functional_enrichment/go_cluster18.png", go_plot_cluster18, width=10, height=10)
```











## Conduct enrichment for cluster developmental patterns 

Conduct at biological process level. Filter for adjusted p-value <0.05.  
```{r}
pattern_list<-levels(as.factor(group_list$pattern))

for (pattern_id in pattern_list) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(filtered_Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(filtered_Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- group_list%>%
      filter(pattern==pattern_id)%>%
      pull(gene)
    
    ##Get a list of GO Terms for all genes detected in our dataset 
    GO.terms <- filtered_Mcap.annot%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.05
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("Mcap2020/Output/TagSeq/GenomeV3/DESeq/functional_enrichment/goseq_", pattern_id, ".csv"))
    
}
```

Create an empty dataframe to populate (only do this once).  

```{r}
pattern_go_results <- data.frame()
```

Run a loop to filter enrichment for each temperature contrast using rrvgo. 

```{r}
for (pattern_id in pattern_list) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("Mcap2020/Output/TagSeq/GenomeV3/DESeq/functional_enrichment/goseq_", pattern_id, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      #filter(numInCat>5)%>%
      arrange(., bh_adjust)
    
    print(pattern_id)
    print(length(unique(go_results$GOterm)))
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #calculate similarity 
    scores <- setNames(-log10(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$pattern<-pattern_id
    
    print(length(unique(go_results$GOterm)))
    
    #add to data frame 
    go_df <- go_results
    pattern_go_results <- rbind(pattern_go_results, go_df) 

}
```

Semantic similarity reduced GO terms for each cluster.    

```{r}
length(unique(pattern_go_results$GOterm))
```
There are 459 total enriched GO terms detected when analyzed by pattern groups.  

## Plot enrichment for each developmental pattern

Early development
```{r}
early_data<-pattern_go_results %>% 
  filter(pattern=="Early")

# First, let's find the unique ParentTerm values
unique_early_terms <- unique(early_data$ParentTerm)

# Determine the number of unique ParentTerm values
num_unique <- length(unique_early_terms)

# Determine the midpoint for splitting into two batches
midpoint <- ceiling(num_unique / 2)
midpoint <- 12

early_data <- early_data %>%
  mutate(batch = ifelse(match(ParentTerm, unique_early_terms) <= midpoint, 1, 2))

early_data$ParentTerm <- factor(early_data$ParentTerm) 
early_data$ParentTerm <- factor(early_data$ParentTerm, levels = rev(levels(early_data$ParentTerm)))

go_plot_earlyA <- early_data %>% 
  filter(batch=="1")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 30, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("GO Term")+
  xlab("Hits (%)")+
  ggtitle("Early Development (1-38 hpf); 1887 genes")+
  theme(legend.position = "none",
        axis.text.y = element_text(size=18),
        axis.title.y = element_text(size=24, face="bold"),
        axis.title.x = element_text(size = 24, face = "bold"),
        plot.title = element_text(size = 24, face = "bold"), 
        strip.text.y.right=element_text(angle=0, size=18), 
        legend.title=element_text(size=22),
        legend.text=element_text(size=18)); go_plot_earlyA

go_plot_earlyB <- early_data %>% 
  filter(batch=="2")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 30, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("")+
  xlab("Hits (%)")+
  ggtitle("")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=18),
        axis.title.y = element_text(size=24, face="bold"),
        axis.title.x = element_text(size = 24, face = "bold"),
        plot.title = element_text(size = 24, face = "bold"), 
        strip.text.y.right=element_text(angle=0, size=18), 
        legend.title=element_text(size=22),
        legend.text=element_text(size=18)); go_plot_earlyB

go_plot_early<-plot_grid(go_plot_earlyA, go_plot_earlyB, ncol=2)

ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/DESeq/functional_enrichment/go_early.png", go_plot_early, width=36, height=36)
```

Mid development
```{r}
go_plot_mid <- pattern_go_results %>% 
  filter(pattern=="Mid")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 40, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("GO Term")+
  xlab("Hits (%)")+
  ggtitle("Mid-Development (65-163 hpf); 1696 genes")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=14),
        axis.title.y = element_text(size=18, face="bold"),
        axis.title.x = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 18, face = "bold"), 
        strip.text.y.right=element_text(angle=0, size=14)); go_plot_mid

ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/DESeq/functional_enrichment/go_mid.png", go_plot_mid, width=12, height=10)
```

Late development 
```{r}
late_data<-pattern_go_results %>% 
  filter(pattern=="Late")

# First, let's find the unique ParentTerm values
unique_late_terms <- unique(late_data$ParentTerm)

# Determine the number of unique ParentTerm values
num_unique <- length(unique_late_terms)

# Determine the midpoint for splitting into two batches
midpoint <- ceiling(num_unique / 2)
midpoint <- 7

late_data <- late_data %>%
  mutate(batch = ifelse(match(ParentTerm, unique_late_terms) <= midpoint, 1, 2))

late_data$ParentTerm <- factor(late_data$ParentTerm) 
late_data$ParentTerm <- factor(late_data$ParentTerm, levels = rev(levels(late_data$ParentTerm)))

go_plot_lateA <- late_data %>% 
  filter(batch=="1")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 20, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("GO Term")+
  xlab("Hits (%)")+
  ggtitle("Late Development (183-231 hpf); 2789 genes")+
  theme(legend.position = "none",
        axis.text.y = element_text(size=14),
        axis.title.y = element_text(size=18, face="bold"),
        axis.title.x = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 18, face = "bold"), 
        strip.text.y.right=element_text(angle=0, size=14)); go_plot_lateA

go_plot_lateB <- late_data %>% 
  filter(batch=="2")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=percHits, y=term, fill=over_represented_pvalue, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 20, multi_line = TRUE))+
  geom_point(aes(group=term))+
  geom_segment(aes(x=0, xend=percHits, y=term, yend=term),linewidth=1) +
  scale_colour_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(name = "Over Rep p-value", colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("")+
  xlab("Hits (%)")+
  ggtitle("")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=14),
        axis.title.y = element_text(size=18, face="bold"),
        axis.title.x = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 18, face = "bold"), 
        strip.text.y.right=element_text(angle=0, size=14)); go_plot_lateB

go_plot_late<-plot_grid(go_plot_lateA, go_plot_lateB, ncol=2)

ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/DESeq/functional_enrichment/go_late.png", go_plot_late, width=24, height=16)
```








