---
title: "GoSeq Functional Annotation of WGCNA gene expression results"
author: "Ariana S Huffmyer"
date: "2023"
output: html_document
editor_options: 
  chunk_output_type: console
---

Functional enrichment of WGCNA modules produced in `WGCNA_Mcap.Rmd` script for the *Montipora capitata* 2020 developmental time series.  This script uses the V3 genome from Cyanophora Rutgers.  

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# The following setting is important, do not omit.
options(stringsAsFactors = FALSE) #Set Strings to character
```

# **Load required libraries**    

```{r}
if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse') 
if ("genefilter" %in% rownames(installed.packages()) == 'FALSE') install.packages('genefilter') 
if ("DESeq2" %in% rownames(installed.packages()) == 'FALSE') install.packages('DESeq2') 
if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer') 
if ("WGCNA" %in% rownames(installed.packages()) == 'FALSE') install.packages('WGCNA') 
if ("flashClust" %in% rownames(installed.packages()) == 'FALSE') install.packages('flashClust') 
if ("gridExtra" %in% rownames(installed.packages()) == 'FALSE') install.packages('gridExtra') 
if ("ComplexHeatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('ComplexHeatmap') 
if ("goseq" %in% rownames(installed.packages()) == 'FALSE') install.packages('goseq') 
if ("dplyr" %in% rownames(installed.packages()) == 'FALSE') install.packages('dplyr') 
if ("clusterProfiler" %in% rownames(installed.packages()) == 'FALSE') install.packages('clusterProfiler') 
if ("pheatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('pheatmap') 
if ("magrittr" %in% rownames(installed.packages()) == 'FALSE') install.packages('magrittr') 
if ("rtracklayer" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rtracklayer")
if ("GenomicRanges" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GenomicRanges")
if ("plyranges" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("plyranges")
if ("GSEABase" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GSEABase")
if ("GOSim" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GOSim")
if ("stats" %in% rownames(installed.packages()) == 'FALSE') install.packages("stats")
if ("ggdendro" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("ggdendro")
#if ("KEGG.db" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("KEGG.db")
if ("GO.db" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GO.db")
if ("rrvgo" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rrvgo")

#BiocManager::install("org.Ce.eg.db", force=TRUE) #install if needed 

library("tidyverse")
library("genefilter")
library("DESeq2")
library("RColorBrewer")
library("WGCNA")
library("flashClust")
library("gridExtra")
library("ComplexHeatmap")
library("goseq")
library("dplyr")
library("clusterProfiler")
library("pheatmap")
library("magrittr")
library("rtracklayer")
library("GenomicRanges")
library("plyranges")
library("GSEABase")
library("GOSim")
library("stats")
library("ggdendro")
#library("KEGG.db")
library("GO.db")
library("rrvgo")
```

# **Read in data**  

Load in Gene Info data frame generated in WGCNA_Mcap_V3.Rmd script and subset list of genes and module assignment for each gene.  

*Important! Gnereate geneInfo.csv before this step! Run the WGCNA_Mcap_V3.Rmd in Scripts to generate the most up to date file. It is a large file that cannot be pushed/updated to Git. 

```{r}
geneInfo<-read.csv("Mcap2020/Output/TagSeq/GenomeV3/geneInfo.csv")
dim(geneInfo)

geneInfo<-geneInfo%>%
  #dplyr::select(gene_id, moduleColor)%>%
  rename(Module=moduleColor)%>%
  mutate(Module=as.factor(Module))

dim(geneInfo)
```

There are 11,475 genes in our Gene Info file.  

Get gene length information.  
```{r}
#import file
gff <- rtracklayer::import("Mcap2020/Data/TagSeq/Montipora_capitata_HIv3.genes.gff3") #if this doesn't work, restart R and try again 

transcripts <- subset(gff, type == "transcript") #keep only transcripts 

transcripts_gr <- makeGRangesFromDataFrame(transcripts, keep.extra.columns=TRUE) #extract length information

transcript_lengths <- width(transcripts_gr) #isolate length of each gene

seqnames<-transcripts_gr$ID #extract list of gene id 

lengths<-cbind(seqnames, transcript_lengths)

lengths<-as.data.frame(lengths) #convert to data frame

```

Add in length to the gene info file .    

```{r}
geneInfo$Length<-lengths$transcript_lengths[match(geneInfo$gene_id, lengths$seqnames)]
```

Format GO terms to remove dashes and quotes.  
```{r}
#remove dashes, remove quotes from columns and separate by semicolons (replace , with ;) in KEGG_ko, KEGG_pathway, and Annotation.GO.ID columns
geneInfo$Annotation.GO.ID <- gsub(",", ";", geneInfo$Annotation.GO.ID)
geneInfo$Annotation.GO.ID <- gsub('"', "", geneInfo$Annotation.GO.ID)
geneInfo$Annotation.GO.ID <- gsub("-", NA, geneInfo$Annotation.GO.ID)
```

# **Run GoSeq and Plot by Module**  

## 1. Run the GOSeq function in a loop by module number to test for GO term enrichment.    

Due to high number of enriched GO terms, I am using an adjusted p-value threshold of <0.001 and using `rrvgo` package to reduce redundancy in list of GO terms.  

```{r}
module_vector<-c(levels(geneInfo$Module))

for (module in module_vector) {
  
    ### Generate vector with names of all genes 
    ALL.vector <- c(geneInfo$gene_id)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(geneInfo$Length)
    
    ### Generate vector with names in just the module we are analyzing
    ID.vector <- geneInfo%>%
      filter(Module==module)%>%
      #get_rows(.data[[module]]))%>%
      pull(gene_id)
    
    ##Get a list of GO Terms for each module
    GO.terms <- geneInfo%>%
      filter(Module==module)%>%
      #filter(get_rows(.data[[module]]))%>%
      dplyr::select(gene_id, Annotation.GO.ID)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$Annotation.GO.ID), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "Annotation.GO.ID")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.0001
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.001) %>%
        dplyr::arrange(., ontology, bh_adjust)
    
    #Write file of results 
    write_csv(GO, file = paste0("Mcap2020/Output/TagSeq/GenomeV3/GOSeq/goseq_module", module, ".csv"))
    
}
```

Run a loop to plot enrichment for each module at each level of ontology.  
```{r}
module_vector<-c(levels(geneInfo$Module))
ontologies<-c("BP", "MF")

#add vector for terms of interest to reduce number of GO terms 
keywords<-c("metabolism", "carbon", "carbohydrate", "lipid", "fatty", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane", "glucose", "organic substance transport", "response to stress", "antioxidant")

for (module in module_vector) {
  
  for (category in ontologies) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("Mcap2020/Output/TagSeq/GenomeV3/GOSeq/goseq_module", module, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
    
      #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #calculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]

    go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
    
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    facet_grid(ParentTerm ~ ., scales = "free_y", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())

ggsave(filename=paste0("Mcap2020/Figures/TagSeq/GenomeV3/GOSeq/GOenrich_Module", module, "_", category, ".png"), plot=GO.plot, dpi=300, width=8, height=49, units="in")
    
  }
}

```

# **Run GoSeq and Plot by Pattern** 

## 1. Identify modules associated with ontogenetic patterns 

Generate a list of modules in each pattern and add identifier to dataframe. 

Identified from eigengene plots. "early" indicates eggs, embryos; "mid" indicate late embryos, larvae; "late" indicates polyps, recruits. 

Increase over development = 12, 3, 5, 9
Decrease over development = 1, 8 
Peak in mid-development = 2, 6

Module 7 only peaks in one stage so we won't include it here. 

```{r}
increase<-c("12", "3", "5", "9")
decrease<-c("1", "8")
mid<-c("2", "6")

geneInfo<-geneInfo%>%
  mutate(Pattern=if_else(Module %in% increase, "Increasing", "NA"))%>%
  mutate(Pattern=if_else(Module %in% decrease, "Decreasing", Pattern))%>%
  mutate(Pattern=if_else(Module %in% mid, "Mid", Pattern))

pattern_vector<-c("Increasing", "Decreasing", "Mid")
```

## 2. Run the GOSeq function in a loop by lifestage patterns to test for GO term enrichment.  

Generate dataframes for each pattern 
```{r}
for (pattern in pattern_vector) {
  
    ### Generate vector with names of all genes 
    ALL.vector <- c(geneInfo$gene_id)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(geneInfo$Length)
    
    ### Generate vector with names in just the module we are analyzing
    ID.vector <- geneInfo%>%
      filter(Pattern==pattern)%>%
      #get_rows(.data[[module]]))%>%
      pull(gene_id)
    
    ##Get a list of GO Terms for each module
    GO.terms <- geneInfo%>%
      filter(Pattern==pattern)%>%
      #filter(get_rows(.data[[module]]))%>%
      dplyr::select(gene_id, Annotation.GO.ID)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$Annotation.GO.ID), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "Annotation.GO.ID")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.0001) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("Mcap2020/Output/TagSeq/GenomeV3/GOSeq/goseq_pattern_", pattern, ".csv"))
    
}
```

Run a loop to plot enrichment for each pattern at each level of ontology.  
```{r}
pattern_vector<-c("Increasing", "Decreasing", "Mid")
ontologies<-c("BP", "MF")

#add vector for terms of interest to reduce number of GO terms 
keywords<-c("metabolism", "carbon", "carbohydrate", "lipid", "fatty", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane", "glucose", "organic substance transport", "response to stress", "antioxidant")

for (pattern in pattern_vector) {
  
  for (category in ontologies) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("Mcap2020/Output/TagSeq/GenomeV3/GOSeq/goseq_pattern_", pattern, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]

    go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))%>%
      filter(!ParentTerm=="multicellular organismal response to stress") #remove category with only 1 go term not of interest
      
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    facet_wrap(~ pattern) +
    scale_size(name="Over rep. p-value", trans="reverse")+
    facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())

ggsave(filename=paste0("Mcap2020/Figures/TagSeq/GenomeV3/GOSeq/GOenrich_Pattern_", pattern, "_", category, ".png"), plot=GO.plot, dpi=300, width=8, height=49, units="in")
    
  }
}

```

## 3. Run the GOseq function in a loop by pattern to test for KEGG term enrichment   

```{r}
pattern_vector<-c("Increasing", "Decreasing", "Mid")

for (pattern in pattern_vector) {
  
    ### Generate vector with names of all genes 
    ALL.vector <- c(geneInfo$gene_id)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(geneInfo$Length)
    
    ### Generate vector with names in just the module we are analyzing
    ID.vector <- geneInfo%>%
      filter(Pattern==pattern)%>%
      #get_rows(.data[[module]]))%>%
      pull(gene_id)
    
    ##Get a list of GO Terms for each module
    KO.terms <- geneInfo%>%
      filter(Pattern==pattern)%>%
      #filter(get_rows(.data[[module]]))%>%
      dplyr::select(gene_id, KEGG)

    ##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)

    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write_csv(KO, file = paste0("Mcap2020/Output/TagSeq/GenomeV3/GOSeq/goseq_KEGG_pattern_", pattern, ".csv"))
    
}
```

Run a loop to plot enrichment for each module at each level of KEGG terms.    
```{r}

for (pattern in pattern_vector) {

    #Read relevant file of results from goseq analysis  
    ko_results<-read_csv(file = paste0("Mcap2020/Output/TagSeq/GenomeV3/GOSeq/goseq_KEGG_pattern_", pattern, ".csv"))
    
    ko_results<-ko_results%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
    
    #plot significantly enriched KO terms 
  KO.plot <-  ggplot(ko_results, aes(x = pattern, y = category)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())

ggsave(filename=paste0("Mcap2020/Figures/TagSeq/GenomeV3/GOSeq/GOenrich_KEGG_Pattern_", pattern, "_", category, ".png"), plot=KO.plot, dpi=300, width=6, height=12, units="in")
    
}

```

Fetch KEGG terms from database: https://www.genome.jp/kegg/ko.html 

KEGG Terms from Increasing over Development: 
ko:K20478    GOLGB1; golgin subfamily B member 1
ko:K22614    NLRC3, NOD3; NLR family CARD domain-containing protein 3
ko:K01672    CA; carbonic anhydrase [EC:4.2.1.1]
ko:K09848    TRAF4; TNF receptor-associated factor 4
ko:K11150    RDH8; retinol dehydrogenase 8 [EC:1.1.1.-]

KEGG Terms from Decreasing over Development:  
ko:K09228    KRAB; KRAB domain-containing zinc finger protein
ko:K11251    H2A; histone H2A
ko:K11252    H2B; histone H2B

KEGG Terms from Mid Development:  
ko:K10408    DNAH; dynein axonemal heavy chain

## 4. Plot all ontogenetic categories together for categories of interest    

Plot enrichment for each pattern together in one plot. Customize first for carbohydrate metabolism related BP ontology.  
```{r}
#bring in dataframes and bind them together 
file1<-read_csv("Mcap2020/Output/TagSeq/GenomeV3/GOSeq/goseq_pattern_Increasing.csv")%>%mutate(pattern="Increasing")
file2<-read_csv("Mcap2020/Output/TagSeq/GenomeV3/GOSeq/goseq_pattern_Decreasing.csv")%>%mutate(pattern="Decreasing")
file3<-read_csv("Mcap2020/Output/TagSeq/GenomeV3/GOSeq/goseq_pattern_Mid.csv")%>%mutate(pattern="Mid")
patterns_df<-rbind(file1, file2, file3)

patterns_df$pattern<-factor(patterns_df$pattern, levels=c("Decreasing", "Mid", "Increasing"))

category<-c("BP")

#add vector for terms of interest to reduce number of GO terms 
carbs<-c("carbohydrate")
  
    #Read relevant file of results from goseq analysis  
    go_results<-patterns_df
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]

    go_results<-go_results%>%
      filter(grepl(paste(carbs, collapse="|"), ParentTerm))
      
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot.carb <-  ggplot(go_results, aes(x = pattern, y = term)) + 
    geom_point(aes(size=numInCat, fill=bh_adjust, colour=bh_adjust)) + 
    scale_size(name="Number of Genes", range=c(5,12))+
    scale_fill_gradient(name = "Adjusted p-value", low="black", high="gray")+
    scale_colour_gradient(name = "Adjusted p-value", low="black", high="gray")+
    facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 6, multi_line = TRUE))+
    xlab("Ontogenetic Pattern")+
    ggtitle("Carbohydrate Metabolism")+
    theme_bw() + 
    theme(panel.border = element_blank(), 
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
          strip.background =element_rect(fill="white"),
          strip.text.y = element_text(angle=0, size = 18, face="bold"),
          strip.text.x = element_text(size = 16),
          axis.text.x = element_text(size = 20, angle=45, hjust=1),
          axis.text.y = element_text(size = 16),
          axis.title.x = element_text(size = 22, face="bold"),
          legend.text=element_text(size=16),
          title=element_text(size=22, face="bold"),
          plot.title = element_text(hjust = 0.1),
          axis.title.y = element_blank())


ggsave(filename=paste0("Mcap2020/Figures/TagSeq/GenomeV3/GOSeq/GOenrich_AllPatterns_Carbohdyrate.png"), plot=GO.plot.carb, dpi=300, width=13, height=18, units="in")

#save object for merging all plots together for manuscript
go_results_carbs<-go_results%>%
  mutate(category="Carbohydrate Metabolism")

```

Next plot for lipid metabolism.  
```{r}
#add vector for terms of interest to reduce number of GO terms 
lipids<-c("lipid", "fatty")
  
    #Read relevant file of results from goseq analysis  
    go_results<-patterns_df
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]

    go_results<-go_results%>%
      filter(grepl(paste(lipids, collapse="|"), ParentTerm))
      
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot.lipid <-  ggplot(go_results, aes(x = pattern, y = term)) + 
    geom_point(aes(size=numInCat, fill=bh_adjust, colour=bh_adjust)) + 
    scale_size(name="Number of Genes", range=c(5,12))+
    scale_fill_gradient(name = "Adjusted p-value", low="black", high="gray")+
    scale_colour_gradient(name = "Adjusted p-value", low="black", high="gray")+
    xlab("Ontogenetic Pattern")+
    ggtitle("Lipid Metabolism")+
    theme_bw() + 
    theme(panel.border = element_blank(), 
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
          strip.background =element_rect(fill="white"),
          strip.text.y = element_text(angle=0, size = 18, face="bold"),
          strip.text.x = element_text(size = 16),
          axis.text.x = element_text(size = 20, angle=45, hjust=1),
          axis.text.y = element_text(size = 16),
          axis.title.x = element_text(size = 22, face="bold"),
          legend.text=element_text(size=16),
          title=element_text(size=22, face="bold"),
          plot.title = element_text(hjust = -0.1),
          axis.title.y = element_blank())

ggsave(filename=paste0("Mcap2020/Figures/TagSeq/GenomeV3/GOSeq/GOenrich_AllPatterns_Lipids.png"), plot=GO.plot.lipid, dpi=300, width=13, height=18, units="in")

#save object for merging all plots together for manuscript
go_results_lipids<-go_results%>%
  mutate(category="Lipid Metabolism")
```

Plot enrichment for each pattern together in one plot. Customize first for transport/communication related BP ontology.  
```{r}
category<-c("BP")

#add vector for terms of interest to reduce number of GO terms 
symbiosis<-c("regulation of cell communication", "transmembrane transport", "transmembrane", "organic substance transport", "antioxidant", "oxygen")
  
    #Read relevant file of results from goseq analysis  
    go_results<-patterns_df
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]

    go_results<-go_results%>%
      filter(grepl(paste(symbiosis, collapse="|"), ParentTerm))
      
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot.sym <-  ggplot(go_results, aes(x = pattern, y = term)) + 
    geom_point(aes(size=numInCat, fill=bh_adjust, colour=bh_adjust)) + 
    scale_size(name="Number of Genes", range=c(5,12))+
    scale_fill_gradient(name = "Adjusted p-value", low="black", high="gray")+
    scale_colour_gradient(name = "Adjusted p-value", low="black", high="gray")+
    xlab("Ontogenetic Pattern")+
    ggtitle("Communication \nand Transport")+
    theme_bw() + 
    theme(panel.border = element_blank(), 
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
          strip.background =element_rect(fill="white"),
          strip.text.y = element_text(angle=0, size = 18, face="bold"),
          strip.text.x = element_text(size = 16),
          axis.text.x = element_text(size = 20, angle=45, hjust=1),
          axis.text.y = element_text(size = 16),
          axis.title.x = element_text(size = 22, face="bold"),
          legend.text=element_text(size=16),
          title=element_text(size=22, face="bold"),
          plot.title = element_text(hjust = 0),
          axis.title.y = element_blank())

ggsave(filename=paste0("Mcap2020/Figures/TagSeq/GenomeV3/GOSeq/GOenrich_AllPatterns_Symbiosis.png"), plot=GO.plot.sym, dpi=300, width=13, height=18, units="in")

#save object for merging all plots together for manuscript
go_results_symbiosis<-go_results%>%
  mutate(category="Communication and Transport")

```

## 5. Plot trends of eigenegenes from modules for each ontogenetic pattern  

Plot eigengene values for each pattern identified above from the module eigengene expression values.  

```{r}
pattern_expression<-read_csv("Mcap2020/Output/TagSeq/GenomeV3/WGCNA_eigengene_expression.csv")

#add pattern identifier to each module 
increase_list<-c("ME12", "ME3", "ME5", "ME9")
decrease_list<-c("ME1", "ME8")
mid_list<-c("ME2", "ME6")

pattern_expression<-pattern_expression%>%
  mutate(Pattern=if_else(Module %in% increase_list, "Increasing", "NA"))%>%
  mutate(Pattern=if_else(Module %in% decrease_list, "Decreasing", Pattern))%>%
  mutate(Pattern=if_else(Module %in% mid_list, "Mid", Pattern))

head(pattern_expression)
```

Summarize number of modules for each pattern and number of genes in each.  
```{r}
pattern_expression%>%
  group_by(Pattern)%>%
  summarise(number_module=length(unique(Module)))
  
#Gene counts from WGCNA analysis 
gene_counts<-read_csv("Mcap2020/Output/TagSeq/GenomeV3/genes_per_module.csv")
names(gene_counts)<-c("Module", "Genes")

#add pattern group for each module  
gene_counts<-gene_counts%>%
  mutate(Pattern=if_else(Module %in% c("12", "3", "5", "9"), "Increasing", "NA"))%>%
  mutate(Pattern=if_else(Module %in% c("1", "8"), "Decreasing", Pattern))%>%
  mutate(Pattern=if_else(Module %in% c("2", "6"), "Mid", Pattern))

gene_counts%>%
  group_by(Pattern)%>%
  summarise(number_genes=sum(Genes))
```

There are 3644 genes (2 modules) decreasing, 3856 (4 modules) increasing, 3676 (2 modules) in mid development, and 299 unassigned. 

```{r}
f_labels <- data.frame(Pattern = c("Increasing across Development", "Peaking in Mid Development", "Decreasing across Development"), label = c("n=4 modules; n=3856 genes", "n=2 modules; n=3676 genes", "n=2 modules; n=3644 genes"))
f_labels
```

Plot patterns.  

```{r}
time_point_order = c("Egg (1 hpf)", "Embryo (5 hpf)", "Embryo (38 hpf)", "Embryo (65 hpf)", "Larvae (93 hpf)", "Larvae (163 hpf)", "Larvae (231 hpf)", "Metamorphosed Polyp (231 hpf)", "Attached Recruit (183 hpf)", "Attached Recruit (231 hpf)") #Set time_point order

#add group to dataframe for scale colour manual 
pattern_expression<-pattern_expression%>%
  mutate(group=if_else(grepl("Egg", lifestage), "Fertilized Egg", as.character(lifestage)))%>%
  mutate(group=if_else(grepl("Embryo", lifestage), "Embryo", as.character(group)))%>%
  mutate(group=if_else(grepl("Larvae", lifestage), "Larvae", as.character(group)))%>%
  mutate(group=if_else(grepl("Polyp", lifestage), "Metamorphosed Polyp", as.character(group)))%>%
  mutate(group=if_else(grepl("Recruit", lifestage), "Attached Recruit", as.character(group)))

list_groups<-c("Fertilized Egg", "Embryo", "Larvae", "Metamorphosed Polyp", "Attached Recruit")
pattern_expression$group<-factor(as.character(pattern_expression$group), levels=list_groups)

pattern_expression$Pattern<-factor(as.character(pattern_expression$Pattern), levels=c("Decreasing", "Mid", "Increasing"))

pattern_plots<-pattern_expression%>%
  group_by(Pattern, lifestage) %>%
  filter(!Pattern=="NA")%>%
  
  ggplot(aes(x=lifestage, y=Mean, group=lifestage)) +
  facet_wrap(~Pattern, nrow=1, ncol=3)+
  geom_point(aes(colour=group), size=6) +
  geom_smooth(aes(group=1), color="black", method='loess')+
  scale_x_discrete(name="", limits=time_point_order) +
  scale_colour_manual(name="Lifestage", values = c("Fertilized Egg"="#8C510A", "Embryo"="#DFC27D", "Larvae"="#80CDC1", "Metamorphosed Polyp"="#003C30", "Attached Recruit"="#BA55D3")) +
  ylab("Mean Eigengene Expression") +
  geom_hline(yintercept = 0, linetype="dashed", color = "grey")+
  theme_bw() + 
  theme(
        #axis.text.x=element_text(angle = 45, hjust=1, size = 12, color="black"), #set x-axis label size
        axis.text.x=element_blank(),
        strip.background=element_rect(fill="white", color="white"),
        strip.text=element_text(size=26, face="bold"),
        axis.title.x=element_text(size = 14), #set x-axis title size
        axis.ticks.x=element_blank(), #No x-label ticks
        #axis.title.y=element_blank(), #No y-axis title
        axis.text.y=element_text(size = 14, colour="black"), #set y-axis label size,
        panel.border = element_rect(color = "black", fill = NA, size = 1), #set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        legend.text = element_text(colour="black", size=24),
        legend.title = element_text(colour="black", size=26, face="bold"),
        legend.position="bottom",
        plot.background=element_blank(),
        plot.title = element_text(size=26)); pattern_plots

ggsave(filename="Mcap2020/Figures/TagSeq/GenomeV3/ontogeny_patterns_expression.png", plot=pattern_plots, dpi=50 , width=16, height=6, units="in")
```


